\section{Discussion}

%B:  at least two surprising outcomes: (1)  D and P on background of C, and (2) SC not as dead of an end as it was in goldberg2010, if P is partitioned out (can be framed as kind of path-dependence, i.e. qIC path different than rho path, or same as (1), so that I and C are less different on the background of D.
We investigated the individual and joint effects of ploidy and breeding system states in diversification of Solanaceae.
Most significantly, we found that the effects of ploidy, inferred as significant when it is the only trait under consideration, are overshadowed by hidden traits, or breeding system when added to models.  
Interestingly, the difference in diversification rates between the diploids and polyploids is negligible on the background of self-compatibility.
In the same vein, we recovered a smaller difference in diversification rates between self-incompatible and self-compatible lineages against the diploid genomic background.
%Inference of transition rates can appreciably across 
Motivated by the widespread findings of paleopolyploid ancestry across angiosperms, we also examined and found modest support for diploidization in the history of the family. %B: "mixed" "not strong" what's the right word here?  E: not "weak", maybe "not strong" or "not nearly as strong as anything else"
Disentangling the patterns of diversification linked to multiple traits is difficult, but our strategy allows tests of intricate hypotheses regarding a heterogenous diversification process.
Below, we discuss these key findings and highlight the most interesting and paradoxical implications.
We additionally provide an outline for broader application of phylogenetic comparative methods in the context of interactions among traits and diversification process. 

\subsection*{Disentangling trait effects on diversification}

% confounding effects of other traits

% - value of approach
Disentangling the patterns of diversification linked to  multiple traits allowed us the opportunity to model and test complex hypotheses about the heterogenous diversification process in Solanaceae.
Species are created and go extinct based on multiple and often highly correlated phenotypes. 
However, estimating the speciation and extinction rates remains challenging. 
By carefully aggregating and curating ploidy and breeding system data for Solanaceae, we showed how  the complex interaction between ploidy and breeding system  can lead to diversification conclusions that differ from the diversification conclusions of models that focus on one trait at a time.

% - what happens if PD only, Hidden states
Previous analyses of polyploidy linked to diversification showed how diploids had greater net diversification rates than polyploids when exploring across multiple clades of the angiosperm phylogeny \citet{mayrose_2011, mayrose_2015}. 
Using the most complete dataset for ploidy in a phylogenetic tree for Solanaceae, we were able to replicate the results found by \citet{mayrose_2011} where polyploids have a slower net diversification compare to diploids.
However, we expanded this study to accommodate  background heterogeneity in the diversification process.
When adding heterogeneity we found that it was more likely that an unobserved trait linked to diploid state was the one leading the net diversification patterns, and that there were some ``second-class"  diploids that were not different from  polyploids  in diversification terms (Figure 2A).
This result made us question whether there was a second trait linked to diploidy that could make a differential in the diversification progress.

%     main finding about our traits
For Solanaceae, looking into self-incompatibility was a logical step.
Previous studies have shown that self-incompatible Solanaceae species have also higher rates of diversification compared to their self-compatible counterparts \citep{goldberg_2012}. 
Again, in our sample when looking only into breeding system linked to the diversification process we were able to replicate the slower net diversification for self-compatible taxa (Figure 2C). When this model was expanded to incorporate hidden states we found again that some self-incompatible taxa have faster net diversification rates than self-compatible taxa. However, there is again a  different class of taxa where those differences are not as striking (Figure 2D).  However, investigating both  both polyploidy and breeding system simultaneously for every species in our sample, allowed us to further investigate why some diploids were quantitatively different than polyploids, or why some self-incompatible taxa would be different in diversification terms from some self-compatible when a second class is not.
In the three-state diversification model ID/CD/CP, we represented the complex interactions between ploidy and breeding system. We found that self-incompatible diploids have faster and positive rates than self-compatible diploids and polyploids, and that the difference between the  rates of net diversification of self-compatible diploids is not as large (Figure 2E) as first found by binary trait diversification models (Figure 2A).
This result is key, it not only aligned with the net diversification results from the D/P+A/B model, where a `	`hidden-trait" seem to be dictating the diversification pattern but it also aligned with the hidden state model I/C+A/B where some class of self-compatible net diversification rate might be overlapping with two net diversification rates of self-incompatible.
% Removed by R. Therefore, we consider that finding a heterogenous result in the hidden trait approaches should be be treated as evidence of a second trait that is necessary to consider.
% Removed by R. Pursuing knowledge of  such trait can result on a clearer picture of the importance of trait linked diversification patterns, but also on a better reconstruction on past events in phylogenies.



%     more generally, not okay to approximate musse with bisse when states are correlated (cf Pyron)

% alpha is always low
%     new supp fig showing how A & B are inferred on the tree?  at least for ID/CD/CP+A/B model
%     maybe we can hypothesize factors that underlie the hidden state (geography?)

% pathways: compare/contrast with robertson2011
%     \rho_I > \rho_C in all!

% Will: I really like the pathways analyses and have a couple comments
%
% 1) Were the "without diversification" and "with diversification" analyses computed
%    using the same MAP transition rate estimates? Transition rates among states will
%    be very different if estimated using a model that is diversification-independent 
%    compared to when estimated with a trait-dependent diversification model. Since the point of 
%    showing these side-by-side seems to be emphasizing the importance of considering trait-dependent
%    diversification, I wonder how different the "without diversification" contributions would be
%    if calculated from transition rates estimated using a diversification-independent model.
%  R- Good point. Emma will clarify
%  E: Yes, good point.  Rosana, do you have results for state-independent diversification?  We could compare with using those rates.
%
% 2) I think stochastic mapping would be another useful perspective on the pathways question
%    (and yeah, its a bit late in the game to propose analyses, so maybe next time!). With stochastic 
%    mapping we could use the entire posterior estimate of the transition rates rather than a point
%    estimate. But more importantly we'd get the relative contributions of the pathways estimated 
%    over the actual Solanceae tree (which has a lot of short interdependent branches rather than 
%    hypothetical long branches). 
%R- It will, it was actually on my code I think for the ID/CD/CP I have the stochastic mapping trees. We can do it for the revision. I think the only model where it could not finish was the ID/CD/CP+A/B (was taking forever and we had the restriction of time from MSI)

\subsection*{Pathways to polyploidy}

% examine SC-only families
% no causal correlation in SSI systems
% paths to dioecy
% path-dependent net-diversification

We find that, in Solanaceae, a diploid lineage is much more likely to take a
one-step  ID$\rightarrow$CP pathway to polyploidy rather than a two-step ID$\rightarrow$CD$\rightarrow$CP pathway.
This result is significant because it has a number of implications for the inference of history of breeding and sexual systems in flowering plants.
First, support for the direct pathway lends credence to the idea that breakdown of SI by WGD---by generation of diploid `heteroalleleic' pollen---may often trigger the evolution of gender dimorphism across angiosperms \citep{miller_2000,roberston_2011}.

%missing deeper discussion of with/without SDD/SID estimates and comparison with Robertson et al. 2011.

% qualifying paragraph for pathways
% "This finding is not without a hitch..."
We implicitly modeled trait changes in both of these pathways as anagenetic, transitions occurring within lineages, without associated speciation events.
However, losses of SI (I$\rightarrow$C transitions) are often associated with cladogenesis in Solanaceae \citep{goldberg_2012}, and polyploidization (D$\rightarrow$P transition) also seem to co-occur with speciation events in a number of other families \citep{zhan_2016,freyman_2017}.
Models that fail to consider such cladogenetic changes may make misleading estimates of anagenetic transition rates.
In the only available data bearing on this problem, \citet{goldberg_2012} found that extending the *SSE framework to allow cladogenetic changes did not substantially affect inference of net diversification rates.
Nevertheless, a distinct possibility remains that allowing for cladogenetic transitions could tip the scales further in favor of the one-step pathway to SC polyploidy (CP), but this remains to be tested in future work.
Specifically, the multi/hidden state SSE-based framework we employed here could be extended to incorporate cladogenetic transitions.
% We could also list %W+%R idea above, for stochastic mapping?
Another useful line of inquiry could examine the path-dependent diversification rates, asking whether diversification rates of polyploid SC lineages differ depending on the path taken.
 
% And, are diversification rates of polyploid SC lineages different if one or other path was taken? \citet{charlesworth1985} % E: This is a great question.  We don't answer it with this round of analyses.  But discuss?
% E: To elaborate: In the IC/CD/CP model, the CP state would be subdivided into two hidden values, corresponding to the influx from the two pathways.  Those two states, CP_X and CP_Y, could have different lambda and mu.

%W: 3) I'll bet allowing cladogenetic transitions would make the one-step ID/CP even more dominant.
%    I added the paragraph below regarding cladogenetic transitions, but it doesn't really fit with
%    the rest of the discussion, so modify or discard as needed:
% E: A small defense of excluding clado is that it didn't affect the diversification results in goldberg2010 vs goldberg2012


\subsection*{Diploidization}

%R: By adding diploidization to models of polyploidy linked to diversification it is possible to recover this complicated scenario and to reconcile genomic evidence with stochastic models. 
%In the cases where ploidy level was assigned by us based chromosome multiplicity at the genus level, estimating diploidization  might be a potential issue due to ploidy misclassification.  

Diploidization is widely considered to be relatively rare, compared with polyploidization \citep{husband_2013}, but flowering plant lineages are thought to have experienced at least one round of polyploidization in their evolutionary history \citep{soltis_2015}.
As a consequence, it is possible that nearly all extant species classified as ``diploid" in our analyses, based on cytogenetic data, are possibly secondary diploids, having undergone both polyploidization and re-diploidization.
Unduly ignoring secondary diploidization would clearly underestimate the rate of transitions from polyploids to diploids, and would likely cause inflated net diversification rates for diploids. % E: first clause seems like a tautology? second clause does not seem obvious, but could at least say that it could affect div rate estimates
We find strong support in models that included diploidization \cref{table:bayesfactors}. However, estimating the rate at which it occurs based on ploidy level classifications is hard and highly uncertain (see parameter $\delta$ in \cref{suppfigure:DP,suppfigure:DPAB,suppfigure:IDCDCP,suppfigure:IDCDCPAB}) and can change the pattern of ancestral reconstructions (\cref{suppfigure:DPnodipasr,suppfigure:DPasr,suppfigure:DPnodipABasr,suppfigure:DPABasr,suppfigure:IDCDCPnodipasr,suppfigure:IDCDCPasr,suppfigure:IDCDCPnodipABasr,suppfigure:IDCDCPABasr}).
%We find weak support for inclusion of diploidization. %cite fig and expand (FIXME).
% R- I corrected the above phrase, we found strong support for the inclusion of diploidization, models that have it are always preferred over models that don't however that parameter is hard to estimate

Genomic evidence indicates that a paleo-polyploidization may have occurred prior to the origin of the family Solanaceae, rendering all current `diploids' secondarily derived. 
First, \citet{ku2000} and \citet{blanc2004} posited that the lineage leading to tomato, \textit{Solanum lycopersicum}, may have experienced one or more WGDs.
A subsequent analysis of synteny between grape and \textit{Solanum} genomes, as well as the distribution between inferred paralogs within genomes of \textit{Solanum} (tomato and potato) each suggested that this lineage experienced a likely round of ancient genome duplication or triplication \citep{tomato2012}. 
The age of the peak of paralog Ks distances, is approximately 71 million years \citep{tomato2012}. 
If this is the case, then all of the genomes may have been subsequently re-diploidized, yielding the common extant base chromosome numbers in this and related families, n=11-12, considered diploid \citep{robertson_2011}. 
%Not unlikely, because Arabidopsis n=5.

%B: incorporate ancestral states in absence of I/C consideration to rag on \delta
%Natural and induced tetraploids from a variety of families, including Fabaceae, Onagraceae, Rosaceae, and Solanaceae, almost invariably lose gametophytic SI (reviewed in \citealt{ramsey_1998,stone_2002,mcclure_2009}).
On the other hand, there is little evidence for occurrence of re-diploidization in the time frame we considered, within the nightshades.
First, the Ks-inferred duplication long pre-dates the origin of the family \citep{sarkinen_2013}. 
The best-supported, most-recent WGD, which may represent a reversion from polyploidy to diploidy, is older than the lineage under consideration in this study.
At best, the species we consider are ancestrally diploid, having re-diploidized after a suspected WGD.
% E: But then how is there SI?! Agreed, I'll cover it.
Second, studies comparing map-based genome synteny within the family find no evidence for recent diploidization \citep{wu_2010a}.
Instead, simple genome re-arrangements appear sufficient to explain chromosomal evolution between a number of species, including all of those in the relatively cytogenetically conserved `x=12' group, which includes tomato, potato, eggplant, pepper, and tobacco.
%B: Once this is re-arranged, and made to fit with the rest of the discussion, add what this has to do with re-diploidization and how to interpret $\delta$.
It is perhaps not surprising that models diploidization in our study recovered wide credible intervals for the diploidization parameter, $\delta$. 
Despite the limitations of assignment of ploidy levels, we found some support for models that include pararmeter $\delta$, meaning that further study may be warranted, especially in other groups.
%Moreover, rare state changes may not be easily recovered. Emma, you had a clever way to summarize this. 
%     meaning of diploidization parameter

% generality of approach
%
% issues
%     irreversibility SI -> SC assumption


\bigskip

\subsection*{Conclusions}
% problems: this system has a big set of advantages - carefully curated (co-)occurrences of two traits, genetic basis, etc. and everything is still contradictory and opaque.

% R-Here I copied the limitations of the models idea but needs to be reorganized to be cohesive % Fix this a lot more please
% This is the idea on how thorough analyses should be done.
Estimating rates of trait linked diversification models is not only a problem of difficult parametric inference (as disscussed in  \citet{rabosky_2010, beaulieu_2015}),but also, a problem of inadequate  ``*SSE'' model specification that can result in misleading inferences when the presence of a second trait has a complex interaction with the focus trait first used to model diversification. We presented a roadmap with a series of analyses that can help scientists to identify if an unknown or unobservable trait is worth pursuing. 
In this roadmap, we have proposed fitting HiSSE models that have effectively shown that some binary states associated with diversification under BiSSE analyses are actually not different in net diversification terms, and instead a hidden state is linked to differences in the diversification rate \citep{beaulieu_2016}.  If that is the case on a system, raising the question of the identity of the hidden state should be considered as the next step in the modeling process. When considering  a second candidate trait, one can model the complex interactions via a MuSSE model, especially in systems where multiple traits are suspected to influence diversification, and where interactions between those traits are well understood. 
Since MuSSE might also suffer from hight type I errors considering more heterogeneity using a hidden state model on top of the multistate SSE is necessary to adequately infer the effect of the complex interactions in the speciation and extinction process.

 %%%%%%%%%%%
 %% Notes to Self %%
 %%%%%%%%%%%

% outline
% - disentangling effects
% - general value of approach
%  		another kind of confounding processes, not only transition and diversification, but "other diversifications" and "other transitions", esp. correlated
%
% - our study key findings and details
% - effects of Ploidy (as %R)
% - trumped by BS
% - no difference on background of SC 
% - discuss data/models and their long/shortcomings: 
%	ClaSSE, will's idea for stochastic mapping
%	irreversibility SI -> SC assumption
% - how to improve / what next
%  (may be useful to study in SC families)
%
% - pathways
% 
%- diploidization
% - rate, effect
% - history in solanaceae
%
% - summary
%
% B: NEW REORG COMMENTS: 


% First, include all straightforward Results and related observations
%
% Then try to square the oval of history of DP and IC. Specifically, it is difficult to know what to make of the delta models. Their inclusion is 'positively' supported, but yields results that are hard to reconcile with other data. This is one of the more interesting tensions of the paper, in the sense that it binds all of the discussion point and something here has to come to a head. 
%Either what we think we know about SI/SC and reversals is wrong, or something about diploidization inference is wrong. Why? It is hard to square that we don't see IP state, almost ever!, but it's supposed to be the ancestral state, and not just in Solanaceae, across eudicots! Doesn't demand a full treatment here, but it's worth pointing out that if the SI/SC inference is wrong, then we must include reversals, qCI, here and elsewhere and it will change all kinds of inferences. Then, we'll have to explain maintenance of trans-generic polymorphism, ha, ha! Otherwise, something is off with the inference delta >> rho in some cases. If so, we'll have to explain paleo-WGD inference, etc. Either way, we fundamentally misunderstand the evolution of one of these two traits.

% including delta makes for some nonsense: ancestral states are all P with rampant recent transitions to D (e.g. Fig. S4/S8. Not backed by any data, including synteny, S-allele trans-polymorphism, or anecdotally bearing experimental data about rates of spontaneous polyploidization and diploidization \citep{husband_2012}.

% PATHWAYS -- mutational opportunity vs lineage transition rate reveals something about selection 
%
% Starting with SI-D, selection favoring establishment of SC-D seems more common than that of SC-P.
% This is because the mutation rates are likely higher to SC-P, but rate 3x lower. "
%Husband2012: The prevalence of polyploidy within plants has led to debate about the influence of genome duplication on net species diversification (Otto and Whitton 2000). Researchers have argued, on one hand, that polyploidy may simply accumulate as a unidirectional mutation with little influence on the rate or trajectory of evolutionary divergence (Stebbins 1971; Meyers and Levin 2006). This idea is based on the premise that ploidy is more likely to increase than decrease in time. Although reversals do occur, the resulting haploids are rela- tively rare and often of reduced fitness and high sterility (Stebbins 1980; Ramsey and Schemske 2002). Alternatively, polyploidy itself may enhance evolutionary diversification leading to increased divergence and species richness (Otto and Whitton 2000). Elevated diversification may occur if polyploidy accelerates adaptive divergence and evolution of reproductive isolation (Otto and Whitton 2000) or reduces species extinction. 
%The presence of intraspecific ploidy diversity has three main causes. First, genome duplication is an ongoing genomic mutation within many natural populations and ploidy polymorphism is therefore expected when poly- ploid formation is common. In fact, Ramsey and Schemske (1998) estimated that polyploids form at the rate of 10 5 per individual per generation, well above the genic mutation rate, and in some cases it may be even higher than that (Husband 2004). Second, depending on the frequency and pathways of polyploid formation, diploids and polyploids may be maintained as genetically homogeneous populations, which are justifiably treated as single taxonomic species. Finally, ploidy polymorphism may arise as an artefact of our imperfect taxonomic approach. Since morphological divergence is often used as a criterion for species delinea- tion, polyploid lineages may not be recognized at the species level, especially in autopolyploids, which often have a close resemblance to their diploid progenitors (Soltis et al. 2007).


