\section{Discussion}

%B:  at least two surprising outcomes: (1)  D and P on background of C, and (2) SC not as dead of an end as it was in goldberg2010, if P is partitioned out (can be framed as kind of path-dependence, i.e. qIC path different than rho path, or same as (1), so that I and C are less different on the background of D.
%
Species are composed of vast assemblages of variable traits, many traits are heritable, and many traits could affect the propensity of species to perish or multiply \citep{lewontin_1970}.
Examining the effects of complex trait combinations on lineage diversification, however, remains challenging.
%
Focusing first on ploidy and then on breeding system, we found that considering each trait in isolation provides an incomplete story.
Considering them together, and in conjunction with another hidden factor, provides a fuller picture of macroevolutionary dynamics within Solanaceae.
We hope our work serves as an example of how phylogenetic comparative methods can be used to disentangle the contributions of interacting traits to heterogeneous lineage diversification, and how to statistically argue for increasing complexity in diversification modeling.

\subsection{Interacting traits and lineage diversification} 

% E: should this results summary be much shorter?
Previous analyses of the effects of ploidy on diversification showed that diploids had greater net diversification rates than polyploids across many angiosperm clades \citep{mayrose_2011, mayrose_2015}. 
We obtained a similar outcome when examining ploidy alone in Solanaceae (\cref{figure:netdivall}A), but a consistent effect of ploidy on diversification was not clear when we added a  hidden factor linked to diversification(\cref{figure:netdivall}B).
%
Previous analyses of breeding system in this family showed that SI species were associated with higher diversification rates than their SC counterparts (\citealt{goldberg_2010}; \cref{figure:netdivall}C).
Our new analyses with a hidden trait showed that this pattern persists even under the presences of an unobserved trait linked to diversification.
Under one of the hidden states, however, SC species experienced positive net diversification (\ie not a dead end; cf.\ \citealt{igic_2013}) that exceeded the diversification rate of SI species under the second hidden state (\cref{figure:netdivall}D).
%
Furthermore, our models of the joint macroevolution of ploidy and breeding system that were statistically preferred revealed how the trait combinations associate with diversification.
We found that the highest overall net diversification rate of diploids was dominated by SI diploids, while SC diploids had a lower diversification rate that overlapped with the net diversification of polyploids, which are also SC (\cref{figure:netdivall}E).
Thus, breeding system seems to provide the dominant explanation of diversification differences, with ploidy providing a secondary effect within SC species.

% These results indicate that the hidden trait in the analysis of one focal trait can in part be explained by the other focal trait.
% more generally, not okay to approximate musse with bisse when states are correlated (cf Pyron)

Throughout our numerous model comparisons, models that included a hidden trait emerged as significantly better than those that did not (\cref{supptable:testaddhidden}). 
This is entirely consistent with the expectation that many processes can affect speciation and extinction, far beyond the one or two traits of particular interest to us.
A natural follow-up question is, what is this hidden trait?
Our results show that to some extent, breeding system functions as a hidden trait in the ploidy-centered analysis, and vice versa (\cref{suppfigure:lumpedDP,suppfigure:lumpedIC}).
But there is also strong statistical support for a hidden trait beyond ploidy and breeding system (\cref{supptable:testaddhidden}).
One temptation is to propose specific possible traits to add into the analysis.
For example, the rapidly-diversifying clade of mostly SC species within \emph{Solanum} suggests that geography may play a role. % Boris, please check!  Australian?  fixme
%     but don't be circular, test in different clade
% alpha is always low: new supp fig showing how A & B are inferred on the tree?  at least for ID/CD/CP+A/B model
%Here Rosana adds the value of adding traits with the lumped models
When analyzing the value of adding more traits to a complex model, our suggestion is to opt for the use of lumped models (\cref{figure:lumped,table:lumped}), and verify that adding a trait is statistically preferred.
In the absence of additional information, the hidden states can be view as merely a statistical trick, providing an easy way to model extra heterogeneity without directly representing a specific trait.
In our analyses, the transition rates between and within hidden traits did not affect the direction of the net diversification difference. However, we found that it is important to allow for the hidden state transition rates to be asymmetrical since those models have a better fit over models that fix the hidden state transition rates to be equal (\cref{suppfigure:asymmetric,supptable:asymmetry}).

Although we fit an extensive set of models in order to relax a variety of their assumptions, we did not explore the process of trait change in conjunction with speciation.
That is, our models all assume anagenetic but not cladogenetic trait evolution.
Phylogenetic methods do exist to separate anagenetic and cladogenetic changes \citep{mayrose_2011, goldberg_2012, magnuson-ford_2012}.
These have been applied to show that losses of SI are often associated with cladogenesis in Solanaceae \citep{goldberg_2012}, and that polyploidization co-occurs with speciation events in other families \citep{zhan_2016, freyman_2017}.
Exploring cladogenetic trait change was outside the scope of our modeling complexity.
Although \citet{goldberg_2012} found that allowing cladogenetic changes did not substantially affect inference of net diversification rates associated with breeding system, future work could test whether this process affects diversification rate estimates with the more complex state spaces of our other models.

\subsection{Pathways to polyploidy}

With evolution predominantly in the direction from diploid to polyploid, and from SI to SC, surviving lineages will tend eventually to become SC polyploids.
We found that in Solanaceae, the pathway to this state is more likely to consist of a single step ($ID \rightarrow CP$) than two steps ($ID \rightarrow CD \rightarrow CP$).
We also showed that although this question focuses on the process of state transitions, its answer is affected by the process of lineage diversification.
We used a simple mathematical approach to investigate the contributions of the two pathways, but future work could use stochastic character mapping to more directly estimate the numbers of each type of transition within the clade.

Macroevolutionary transition rates represent a combination of time spent waiting for individuals to arise with a new trait value, and for that new trait value to become widespread within the species.
For our traits, this consists of mutations that break SI or generate polyploid individuals, and selective pressures that cause loss or fixation of these mutants.
Estimates of mutation rates are highly uncertain, but the chance of breakdown of SI within diploids is perhaps $10^{-5}$ per pollen grain; this includes breakdown by tetraploidization \citep{lewis1979}, which is itself estimated to occur at about the same rate \citep{ramsey_1998}.  % unit check?
% Seemingly, then, the simple genic mutation rate that leads to loss of SI is at best equal to tetraploidization mutation rate, and possibly far lower.
In contrast, we estimate a macroevolutionary transition rate from $ID$ to $CD$ that is three-fold greater than the rate from $ID$ to $CP$, indicating that selection restricts the fixation of new polyploids more than of new SC mutants \citep{robertson_2011}. % todo: check if new rates are still 3x

Our findings prompt several further questions about the macroevolutionary pathways of ploidy and breeding system.
%
First, our support for the direct pathway is consistent with the idea that breakdown of SI by whole genome duplication---via diploid `heteroalleleic' pollen---may often trigger the evolution of gender dimorphism as a different mechanism of inbreeding avoidance \citep{miller_2000}.
A further test of this hypothesis would additionally examine the propensity of polyploids generated through either pathway to become dioecious \citep{robertson_2011}.
%
Second, we might wonder whether the propensity for a polyploid species to diversify depends on whether it arose via the one-step or two-step pathway.
This could be tested with a different form of a hidden state model, in which the polyploid state is subdivided based on the pathway taken into that state.
This would be similar to what we did for our pathways analysis, but it would also include the possibility of different diversification rates in those $CP$ substates.
%
Third, the generality of our findings in other families remains to be assessed.
An identical procedure could be used in other families with gametophytic SI.
In clades with sporophytic self-incompatibility systems, however, SI is not disabled by whole genome duplication, so there is no one-step pathway. % ref?
The correlation between breeding system and ploidy may therefore be different, and it is currently unknown whether one of the two-step pathways might dominate.

% pathways: compare/contrast with robertson2011
%     \rho_I > \rho_C in all!
%     likely way to lose SI
%     likely way to become P
%     (compare those relative rates with robertson2011)

\subsection*{Diploidization}

Polyploidization is known to be common in plants, but the commonness of the reverse process---diploidization---is under active investigation \citep{dodsworth_2015,mandakova_2018}.
%
Ignoring diploidization if it is common, could cause us  for models that ignore it to underestimate the polyploidization rate and overestimate of the diploid diversification rate leading to incorrect conclusions.
Therefore, we included a slate of models with a diploidization parameter, and they showed that our main conclusions are robust to this process (\cref{suppfigure:alldip}).
These models also suggested modest statistical support for diploidization occuring within Solanaceae (\cref{supptable:testdiploidization}), although our estimates of the diploidization rate were highly uncertain.
Furthermore, additional lines of evidence for classifying species as diploid or polyploid (beyond the genus chromosome multiplicity that we primarily relied on) would be needed for more reliable conclusions from an analysis such as ours.
%In the cases where ploidy level was assigned by us based chromosome multiplicity at the genus level, estimating diploidization might be a potential issue due to ploidy misclassification.  
%R: By adding diploidization to models of polyploidy linked to diversification it is possible to recover this complicated scenario and to reconcile genomic evidence with comparative phylogenetic models. 
%B: yielding to %R here. I still think there is overwhelming support for this statement, and few of the cited papers show convincingly that, for example, diploidization happens within five orders of magnitude compared with polyploidization, rendering "relatively rare" obsolete or invalid. (P.S. Rare not the same as unimportant!)
%B: was: Diploidization is widely considered to be relatively rare, compared with polyploidization \citep{husband_2013}, but flowering plant lineages are thought to have experienced at least one round of polyploidization in their evolutionary history \citep{soltis_2015}.

% E: Boris, could you please revisit the following two paragraphs?  TODO
% Two questions:
%    Did my reorganization of this section mess anything up?
%    Can this text be shorter?  It's seems disproportionately long, and I'm hesitant to revise it heavily myself because the logical connections aren't entirely clear to me.

Other lines of evidence about the prevalence of diploidization within Solanaceae or its ancestors are mixed or even conflicting.
% recent work on WGDs conflict with other lines of evidence, especially those concerning the observed simple synteny of genomes in Solanaceae, and patterns of evolution at self-incompatibility loci
%
% Part I: evidence for WGD in/near family
On the one hand, polyploidy may have occurred prior to the origin of Solanaceae, rendering all extant `diploids' secondarily derived.
\Citet{ku2000} and \citet{blanc2004} posited that the lineage leading to tomato, \emph{Solanum lycopersicum}, may have experienced one or more whole genome duplications.
A subsequent analysis of synteny between grape and \emph{Solanum} genomes, as well as the distribution between inferred paralogs within genomes of \emph{Solanum} (tomato and potato) each suggested that this lineage experienced a likely round of ancient genome duplication or triplication \citep{tomato2012}. 
The age of the peak of paralog Ks distances is approximately $71 \pm 19$ My \citep{tomato2012}. 
If this is the case, then all of the genomes may have been subsequently diploidized, yielding the widespread and common chromosome numbers in this and related families, n=11--12, presently considered to be diploid \citep{robertson_2011}. 

% Part II: evidence against WGD in/near family
On the other hand, there is little evidence for the occurrence of diploidization within Solanaceae itself.
First, the Ks-inferred duplication \citep{tomato2012} likely pre-dates the origin of the family (49 My, HPD 46--53 My; \citealt{sarkinen_2013}). 
Thus, the species in our analyes could be ancestrally polyploid, but would have diploidized before the root of the family.
Second, studies comparing map-based genome synteny within the family find no evidence for recent diploidization \citep{wu_2010a}.
Instead, simple genome re-arrangements appear sufficient to explain chromosomal evolution between a number of species. % including all of those in the relatively cytogenetically conserved x=12 group, which includes tomato, potato, eggplant, pepper, and tobacco.
Specifically, chromosome numbers and genome comparisons within the family (esp.\ the x=12 clade containing \textit{Solanum} and \textit{Nicotiana}) reveal strong conservation.
\citet{wu_2010a} review the evidence from map-based genome comparisons and find that tomato and potato differ by six inversions, tomato and eggplant by 24 inversions and five translocations, tomato and pepper by 19 inversions and six translocations, and tomato and tobacco by ca.\ 10 inversions and 11 translocations (likely underestimated).
Recovery of such simple relationships would require outstanding convergent loss of duplicated segments.
%
% Part IIb: evidence from SI [might need more explaining?]
Furthermore, whole genome duplication in a eudicot lineage ancestral to Solanaceae would clash with the evidence that the homologous mechanism of SI that has been present continually in many families \citep{igic_2006} breaks down nearly invariably in natural and induced tetraploids \citep{stone_2002, mcclure_2009}.
Most problematically in this context, it is unclear how to explain the maintenance of trans-generic polymorphism at the orthologous S-loci in Solanaceae and other families if SI was previously broken down by polyploidization.
%
Regardless of whether genome polypoidization, followed by widespread diploidization, is a dominant mechanism of genome evolution in Solanaceae, it is clear that more work is needed for a complete understanding of the joint evolution of ploidy and breeding systems.

% Flowering plant lineages are thought to have experienced at least one round of polyploidization in their evolutionary history \citep{soltis_2015}. 
% Following polyploidization, it is possible for genome reorganization, downsizing, and loss to occur \citep{dodsworth_2015, zenil_2016, mandakova_2018}.
% As a consequence, nearly all extant species classified as ``diploid" in our analyses are possibly secondary diploids, having undergone both polyploidization and subsequent diploidization.

% no discussion of magnitude of \delta vs \rho!

% problems: this system has a big set of advantages - carefully curated (co-)occurrences of two traits, genetic basis, etc. and everything is still contradictory and opaque.

\section{Conclusion}

% E: The text previously here was great, but it read to me more like an abstract or methods/results.  I tried to make it more like a big-picture summary.  What other general points do we want to hit here?

Heterogeneity in lineage diversification across time and clades is the rule, rather than the exception.
This background heterogeneity makes it difficult to test for the association of any one, isolated trait with different rates of speciation or extinction.
Our study provides an example of how diversification linked to a particular trait can be better assessed by a suite of more inclusive models that allow for alternative explanations---whether other traits or unknown factors.
Additionally, our analysis of evolutionary pathways to polyploidy shows the importance of including diversification effects even when addressing questions that focus on trait evolution.
