\section{Discussion}

% - value of approach
The present work shows the importance of considering the trait linked diversification patterns under a multivariate approach.
Species are created and go extinct based on multiple and often highly correlated phenotypes, understanding the speciation and extinction processes requires understanding of the evolutionary consequences that those trait correlations produce in organisms.
In the present work we show how considering both polyploidy and breeding system can disentangle the importance (or the lack of) of polyploidy when confronted with the evidence brought by breeding system.

% - what happens if PD only, Hidden states
Using the most complete dataset for polyploidy in a phylogenetic tree in Solanaceae, we were able to replicate the results found by \citet{mayrose_2011}, polyploids have a slower net diversification compare to diploids.
Furthermore,  we also found polyploids had a high probability of having negative diversification which implies that polyploids can become a macroevolutionary dead-end, a result that was also found in the two large angiosperm diversification studies  \citet{mayrose_2011} and \citet{mayrose_2015}.
However, we expanded this study to accommodate  background heterogeneity in the diversification process.
When adding heterogeneity we found that it was more likely that an unobserved trait linked to diploid state was the one leading the net diversification patterns, and that there were some ``second-class"  diploids that were not different from  polyploids  in diversification terms (Figure 2A).
This result lead us to our central question: \textit{what is that other trait linked diploids that makes them different in the diversification process? }.

%     main finding about our traits
In Solanaceae, our immediate intuition was to look into breeding system.
Previous studies shown that self-incompatible Solanaceae species have also higher rates of diversification compared to their self-compatible counterparts \citep{goldberg_2012}.
Self-incompatible species are diploid in our sample and also expected to be diploid due to ... (citation?).
By considering both polyploidy and breeding system simultaneously for every species in our sample, it was possible to disentangle why some diploids were quantitatively different than polyploids.
In the three-state diversification model ID/CD/CP, we found that self-incompatible diploids have faster and positive rates than self-compatible diploids and polyploids, and that the difference between the  rates of net diversification of self-compatible diploids is not as large (Figures 2C) as first found by binary trait diversification models (Figures 2A).
This result is important, since it aligns with the net diversification results of the  D/P+A/B model, where a `	`hidden-trait" seem to be dictating the diversification pattern.
By adding breeding system, we were able to hint at which that hidden-trait possibly is.
Therefore, we consider that finding a heterogenous result in the hidden trait approaches should be be treated as evidence of a second trait that is necessary to consider.
Pursuing knowledge of  such trait can result on a clearer picture of the importance of trait linked diversification patterns, but also on a better reconstruction on past events in phylogenies.



% - diversification  (frame as no-delta first)


%     more generally, not okay to approximate musse with bisse when states are correlated (cf Pyron)

% alpha is always low
%     new supp fig showing how A & B are inferred on the tree?  at least for ID/CD/CP+A/B model
%     maybe we can hypothesize factors that underlie the hidden state (geography?)

% pathways: compare/contrast with robertson2011
%     \rho_I > \rho_C in all!

% Will: I really like the pathways analyses and have a couple comments
%
% 1) Were the "without diversification" and "with diversification" analyses computed
%    using the same MAP transition rate estimates? Transition rates among states will
%    be very different if estimated using a model that is diversification-independent 
%    compared to when estimated with a trait-dependent diversification model. Since the point of 
%    showing these side-by-side seems to be emphasizing the importance of considering trait-dependent
%    diversification, I wonder how different the "without diversification" contributions would be
%    if calculated from transition rates estimated using a diversification-independent model.
%
% 2) I think stochastic mapping would be another useful perspective on the pathways question
%    (and yeah, its a bit late in the game to propose analyses, so maybe next time!). With stochastic 
%    mapping we could use the entire posterior estimate of the transition rates rather than a point
%    estimate. But more importantly we'd get the relative contributions of the pathways estimated 
%    over the actual Solanceae tree (which has a lot of short interdependent branches rather than 
%    hypothetical long branches). 
%
% 3) I'll bet allowing cladogenetic transitions would make the one-step ID/CP even more dominant.
%    I added the paragraph below regarding cladogenetic transitions, but it doesn't really fit with
%    the rest of the discussion, so modify or discard as needed:
Our work shows that a diploid Solanaceae lineage is much more likely to take a
one-step ID/CP pathway to polyploidy rather than a two-step ID/CD/CP pathway. 
We modeled both of these pathways as anagenetic character changes occurring within a species.
However \citet{goldberg_2012} showed that I/C transitions are often associated with speciation events, and similarly \citet{freyman_2017} demonstrated that D/P transitions may also be associated with speciation events. 
Models that fail to consider transitions that occur at speciation (cladogenetic changes) may
make misleading estimates of anagenetic transition rates.
The HiSSE-based models introduced in our work here could be extended to incorporate cladogenetic transitions. 
It is possible that allowing for cladogenetic transitions may enable the one-step pathway
to even further dominate over the two-step pathway to polyploidy, but this remains to be tested in future work.


% where WGD is in Solanaceae--motivation and delta rate analyses
% diploidization: comment
Diploidization needs to be considered in diversification models because higher rates of net diversification (speciation minus extinction) in diploids can be obtained from models that ignore the possibility of a polyploid lineage that has diploidized being the diversification enhancer. Hence, in a diploidization lacking model, diploids will show higher rates of diversification when in fact it is a polyploidy event and its subsequent diploidization that generated higher speciation and less extinction.  By adding diploidization to models of polyploidy linked to diversification it is possible to recover this complicated scenario and to reconcile genomic evidence with stochastic models. 

% B: This is my pitch for motivation of the \delta paragraph and/or wording for how to reconcile estimates and genomic data (providing summary of data--to date)
Diploidization is widely considered to be relatively rare, compared with polyploidization, but flowering plant lineages are thought to have experienced at least one round of polyploidization in their evolutionary history.
As a consequence, it is possible that nearly all extant species classified as ``diploid" in our analyses are possibly secondary diploids, having undergone both polyploidization and re-diploidization.
Genomic evidence indicates that such an event may have occurred prior to the origin of the family Solanaceae. 
\citet{ku2000} and \citet{blanc2004} posited that the lineage leading to tomato, \textit{Solanum lycopersicum}, may have experienced one or more paleopolyploid events.
A subsequent analysis of synteny between grape and \textit{Solanum} genomes, as well as the distribution between inferred paralogs within \textit{Solanum} (tomato and potato) genomes both suggest that this lineage experienced a likely round of ancient genome duplication or triplication \citep{tomato2012}. 
The age of the peak of paralog Ks distances, is approximately 71 million years \citep{tomato2012}. 
If this is the case, then all of the genomes may have been subsequently re-diploidized, because common base chromosome numbers this and related families are n=11-12. 
%Not unlikely, because Arabidopsis n=5.
On the other hand, studies comparing map-based genome synteny within the family find no evidence for recent polyploidization-diploidization \citep{wu_2010a}. 
Simple genome re-arrangements appear sufficient to explain chromosomal evolution between a number of species, including all of those in the relatively cytogenetically conserved `x=12' group, which includes tomato, potato, eggplant, pepper, and tobacco.
%B: Once this is re-arranged, and made to fit with the rest of the discussion, add what this has to do with re-diploidization and how to interpret $\delta$.

%     effect on inferred diversification


% generality of approach
%
% issues
%     meaning of diploidization parameter
%     irreversibility SI -> SC assumption
