\section{Discussion}

% outline
% - disentangling effects
% - general value of approach
%  		another kind of confounding processes, not only transition and diversification, but "other diversifications" and "other transitions", esp. correlated
%
% - our study key findings and details
% - effects of Ploidy (as %R)
% - trumped by BS
% - no difference on background of SC 
% - discuss data/models and their long/shortcomings: 
%	ClaSSE, will's idea for stochastic mapping
%	irreversibility SI -> SC assumption
% - how to improve / what next
%  (may be useful to study in SC families)
%
% - pathways
% 
%- diploidization
% - rate, effect
% - history in solanaceae
%
% - summary

%B:  at least two surprising outcomes: (1)  D and P on background of C, and (2) SC not as dead of an end as it was in goldberg2010, if P is partitioned out (can be framed as kind of path-dependence, i.e. qIC path different than rho path, or same as (1), so that I and C are less different on the background of D.
We investigated the individual and joint effects of ploidy and breeding system states in diversification of Solanaceae.
Most significantly, we found that the effects of ploidy, inferred as significant when it is the only trait under consideration, are trumped by breeding systems when the two traits are considered jointly.
Interestingly, the difference in diversification rates between the diploids and polyploids is negligible on the background of self-compatibility.
%B: should we mention here that the hidden states presage these results in each case? Maybe we don't want to make too much of that, but it is fairly interesting. Also possibly hit that in your paragraph(s), below.
In the same vein, we recovered a smaller difference in diversification rates between self-incompatible and self-compatible lineages against diploid genomic background. %B: "in diploids" may work better?
%B: need help with pathway single-sentence summary
Motivated by the widespread findings of paleopolyploid ancestry across angiosperms, we also examined support for diploidization in the history of the family, which appears to be weak. %B: "mixed" "not strong" what's the right word here?
Disentangling the patterns of diversification linked to multiple traits is difficult, but our strategy allows tests complex hypotheses regarding a heterogenous diversification process.
Below, we discuss these key findings and outline a broad approach for application of phylogenetic comparative methods in the context of interactions among traits and diversification process. 


% - value of approach
Disentangling the patterns of diversification linked to  multiple traits allowed us the opportunity to model and test complex hypotheses about the heterogenous diversification process in Solanaceae.
Species are created and go extinct based on multiple and often highly correlated phenotypes. However, estimating the speciation and extinction rates remains challenging. Estimating rates of trait linked diversification models is not only a problem of difficult parametric inference (as disscussed in  \citet{rabosky_2010, beaulieu_2015}),but also, a problem of inadequate model specification that can result in misleading inferences when the presence of a second trait has a complex interaction with the focus trait first used to model diversification. We presented a roadmap with a series of analyses that can help scientists to identify if an unknown or unobservable trait is worth pursuing. By carefully aggregating and curating ploidy and breeding system data for Solanaceae, we showed how  the complex interaction between ploidy and breeding system  can lead to diversification conclusions that differ from the diversification conclusions of models that focus on one trait at a time.

% - what happens if PD only, Hidden states
Previous analyses of polyploidy linked to diversification showed how diploids had greater net diversification rates than polyploids when exploring across multiple clades of the angiosperm phylogeny \citet{mayrose_2011, mayrose_2015}. 
Using the most complete dataset for ploidy in a phylogenetic tree for Solanaceae, we were able to replicate the results found by \citet{mayrose_2011} where polyploids have a slower net diversification compare to diploids.
However, we expanded this study to accommodate  background heterogeneity in the diversification process.
When adding heterogeneity we found that it was more likely that an unobserved trait linked to diploid state was the one leading the net diversification patterns, and that there were some ``second-class"  diploids that were not different from  polyploids  in diversification terms (Figure 2A).
This result made us question whether there was a second trait linked to diploidy that could make a differential in the diversification progress.

%     main finding about our traits
For Solanaceae, looking into self-incompatibility was a logical step.
Previous studies have shown that self-incompatible Solanaceae species have also higher rates of diversification compared to their self-compatible counterparts \citep{goldberg_2012}. 
Again, in our sample when looking only into breeding system linked to the diversification process we were able to replicate the slower net diversification for self-compatible taxa (Figure 2C). When this model was expanded to incorporate hidden states we found again that some self-incompatible taxa have faster net diversification rates than self-compatible taxa. However, there is again a  different class of taxa where those differences are not as striking (Figure 2D).  However, investigating both  both polyploidy and breeding system simultaneously for every species in our sample, allowed us to further investigate why some diploids were quantitatively different than polyploids, or why some self-incompatible taxa would be different in diversification terms from some self-compatible when a second class is not.
In the three-state diversification model ID/CD/CP, we represented the complex interactions between ploidy and breeding system. We found that self-incompatible diploids have faster and positive rates than self-compatible diploids and polyploids, and that the difference between the  rates of net diversification of self-compatible diploids is not as large (Figure 2E) as first found by binary trait diversification models (Figure 2A).
This result is key, it not only aligned with the net diversification results from the D/P+A/B model, where a `	`hidden-trait" seem to be dictating the diversification pattern but it also aligned with the hidden state model I/C+A/B where some class of self-compatible net diversification rate might be overlapping with two net diversification rates of self-incompatible.
% Removed by R. Therefore, we consider that finding a heterogenous result in the hidden trait approaches should be be treated as evidence of a second trait that is necessary to consider.
% Removed by R. Pursuing knowledge of  such trait can result on a clearer picture of the importance of trait linked diversification patterns, but also on a better reconstruction on past events in phylogenies.

%     more generally, not okay to approximate musse with bisse when states are correlated (cf Pyron)

% alpha is always low
%     new supp fig showing how A & B are inferred on the tree?  at least for ID/CD/CP+A/B model
%     maybe we can hypothesize factors that underlie the hidden state (geography?)

% pathways: compare/contrast with robertson2011
%     \rho_I > \rho_C in all!

% Will: I really like the pathways analyses and have a couple comments
%
% 1) Were the "without diversification" and "with diversification" analyses computed
%    using the same MAP transition rate estimates? Transition rates among states will
%    be very different if estimated using a model that is diversification-independent 
%    compared to when estimated with a trait-dependent diversification model. Since the point of 
%    showing these side-by-side seems to be emphasizing the importance of considering trait-dependent
%    diversification, I wonder how different the "without diversification" contributions would be
%    if calculated from transition rates estimated using a diversification-independent model.
%  R- Good point. Emma will clarify
%
% 2) I think stochastic mapping would be another useful perspective on the pathways question
%    (and yeah, its a bit late in the game to propose analyses, so maybe next time!). With stochastic 
%    mapping we could use the entire posterior estimate of the transition rates rather than a point
%    estimate. But more importantly we'd get the relative contributions of the pathways estimated 
%    over the actual Solanceae tree (which has a lot of short interdependent branches rather than 
%    hypothetical long branches). 
%R- It will, it was actually on my code I think for the ID/CD/CP I have the stochastic mapping trees. We can do it for the revision. I think the only model where it could not finish was the ID/CD/CP+A/B (was taking forever and we had the restriction of time from MSI)

% 3) I'll bet allowing cladogenetic transitions would make the one-step ID/CP even more dominant.
%    I added the paragraph below regarding cladogenetic transitions, but it doesn't really fit with
%    the rest of the discussion, so modify or discard as needed:
Our work shows that a diploid Solanaceae lineage is much more likely to take a
one-step ID/CP pathway to polyploidy rather than a two-step ID/CD/CP pathway. 
We modeled both of these pathways as anagenetic character changes occurring within a species.
However \citet{goldberg_2012} showed that I/C transitions are often associated with speciation events, and similarly \citet{freyman_2017} demonstrated that D/P transitions may also be associated with speciation events. 
Models that fail to consider transitions that occur at speciation (cladogenetic changes) may
make misleading estimates of anagenetic transition rates.
The HiSSE-based models introduced in our work here could be extended to incorporate cladogenetic transitions. 
It is possible that allowing for cladogenetic transitions may enable the one-step pathway
to even further dominate over the two-step pathway to polyploidy, but this remains to be tested in future work.

% And, are diversification rates of polyploid SC lineages different if one or other path was taken? \citet{charlesworth1985} % E: This is a great question.  We don't answer it with this round of analyses.  But discuss?

Diploidization needs to be considered in diversification models because higher rates of net diversification (speciation minus extinction) in diploids can be obtained from models that ignore the possibility of a polyploid lineage that has diploidized being the diversification enhancer. 
Hence, in a diploidization lacking model, diploids will show higher rates of diversification when in fact it is a polyploidy event and its subsequent diploidization that generated higher speciation and less extinction.  
By adding diploidization to models of polyploidy linked to diversification it is possible to recover this complicated scenario and to reconcile genomic evidence with stochastic models. 
In the cases where ploidy level was assigned by us based chromosome multiplicity at the genus level, estimating diploidization  might be a potential issue due to ploidy misclassification.  


%B: v2.0
\subsection*{Diploidization}

Diploidization is widely considered to be relatively rare, compared with polyploidization, but flowering plant lineages are thought to have experienced at least one round of polyploidization in their evolutionary history. %cite
As a consequence, it is possible that nearly all extant species classified as ``diploid" in our analyses, based on cytogenetic data, are possibly secondary diploids, having undergone both polyploidization and re-diploidization.
Unduly ignoring secondary diploidization would clearly underestimate the rate of transitions from polyploids to diploids, and would likely cause inflated net diversification rates for diploids.
We find weak support for inclusion of diploidization. %cite fig and expand (FIXME).

Genomic evidence indicates that a paleo-polyploidization may have occurred prior to the origin of the family Solanaceae, rendering all current `diploids' secondarily derived. 
First, \citet{ku2000} and \citet{blanc2004} posited that the lineage leading to tomato, \textit{Solanum lycopersicum}, may have experienced one or more WGDs.
A subsequent analysis of synteny between grape and \textit{Solanum} genomes, as well as the distribution between inferred paralogs within genomes of \textit{Solanum} (tomato and potato) each suggested that this lineage experienced a likely round of ancient genome duplication or triplication \citep{tomato2012}. 
The age of the peak of paralog Ks distances, is approximately 71 million years \citep{tomato2012}. 
If this is the case, then all of the genomes may have been subsequently re-diploidized, yielding the common extant base chromosome numbers in this and related families, n=11-12, considered diploid \citep{robertson_2011}. 
%Not unlikely, because Arabidopsis n=5.

On the other hand, there is little evidence for occurrence of re-diploidization in the time frame we considered, within the nightshades.
First, the Ks-inferred duplication long pre-dates the origin of the family \citep{sarkinen_2013}. 
The best-supported, most-recent WGD, which may represent a reversion from polyploidy to diploidy, is older than the lineage under consideration in this study.
At best, the species we consider are ancestrally diploid, having re-diploidized after a suspected WGD.
Second, studies comparing map-based genome synteny within the family find no evidence for recent diploidization \citep{wu_2010a}.
Instead, simple genome re-arrangements appear sufficient to explain chromosomal evolution between a number of species, including all of those in the relatively cytogenetically conserved `x=12' group, which includes tomato, potato, eggplant, pepper, and tobacco.
%B: Once this is re-arranged, and made to fit with the rest of the discussion, add what this has to do with re-diploidization and how to interpret $\delta$.
It is perhaps not surprising that models diploidization in our study recovered wide credible intervals for the diploidization parameter, $\delta$. 
Despite the limitations of assignment of ploidy levels, we found some support for models that include pararmeter $\delta$, meaning that further study may be warranted, especially in other groups.
%Moreover, rare state changes may not be easily recovered. Emma, you had a clever way to summarize this. 
%     meaning of diploidization parameter

% generality of approach
%
% issues
%     irreversibility SI -> SC assumption

