\begin{quote}
\em{``Among life history traits, reproductive characters that determine mating system patterns are perhaps the most influential in governing macroevolution." 
}
\\
\hspace*{\fill}\rm{ Barrett et al. (1996)}
\end{quote}

\section{Introduction}

%NOTE: please separate each line with a line break!

%B->%E: instead of a major reorganization, I added a breeding system quote from Barrett that presages our results. How's that? It seemed daunting to reorganize ploidy-first to BS-first.  E: nice!

Species accumulate across the tree of life at different rates. 
A possible explanation for this phenomenon is that species possess various traits or character states that differentially affect rates of diversification. 
Dramatic increases in the available phylogenetic and phenotypic data, as well as methodological advances, have greatly accelerated the search for traits that influence diversification traits.
Nevertheless, identifying focal traits associated with rates of speciation and extinction remains a difficult statistical problem \citep[\eg][]{maddison_2015, rabosky_2015, moore_2016, fitzjohn_2009, goldberg_2012, beaulieu_2016, rabosky_2017}. %is it only a "statistical" problem?
In part, the difficulties arise because speciation and extinction may not depend on a single trait, and the context in which traits occur can lead to complex interactions resulting in a heterogenous speciation and extinction rates \citep{beaulieu_2016, caetano_2018, herrera_2018}.
Consequently, examining the association of only one trait with diversification patterns can be misleading. 

Polyploidy, the acquisition of extra sets of chromosomes, is a remarkably common mutation in plants \citep{husband_2013, zenilferguson_2017}.
%R- Polyploid people are careful at not using interchangeably WGD so I switched back to polyploidy terminology. Since we are not using anything molecular I'd rather keep it as polyploidy
The widespread prevalence of variation in ploidy has long been considered a salient feature of flowering plant lineages \citep{stebbins1938}. 
Polyploidy events have the potential to affect many phentoypes and a variety of evolutionary \citep{ramsey_2002} and ecological processes \citep{sessa_2019}.
Polyploids have been thought to have lower rates of net diversification rate than diploids \citep{mayrose_2011, mayrose_2015}. 
Recent studies, however, find common and numerous paleo-polyploidizations, including some preceding the emergence of highly diverse plant clades \citep{soltis_2014, landis_2018}, suggesting that polyploidy events have played an important macroevolutionary role driving innovation and diversification in plants.
Evidence of paleo-polyploidy within the genomes of diploid extant plants also implies the pervasiveness of diploidization, the return of polyploids to the diploid state, throughout the angiosperm phylogeny \citep{soltis_2015, dodsworth_2015}.
% Testing for the presence of diploidization in the context of diversification is important because the presence of a positive net diversification rate for diploids can be the result of an ancestrally polyploid lineage.
% Under this scenario, polyploidy would no longer be considered to reliably slow down diversification, but instead, polyploidy would be the initiator of the diversification process. %B: I think this should read "...polyploidy is no longer be considered an evolutionary dead end."  %R- This last phrase is from Itay and I rather like it. <-- %B: Hmm. But how would it be considered an "initiator of the diversification process?" It seems to be associated with lower diversification. In other words, I also agree with both %R and Itay, but not with what the current phrasing is asserting.
% E: I wrote the two sentences below to try to reconcile the text and comments below.  But then I realized we don't actually test this hypothesis.
% One hypothesis is that diploids have a higher net diversification rate if they were ancestrally polyploid than if they had no polyploid ancestry.
% Testing this hypothesis requires incorporating diploidization in analyses of diversification.
% E: so try again -- okay? 
%R- yup- good resolution
This could affect interpretation of different net diversification rates for diploids and polyploids, especially if diploids have a higher net diversification rate if they were ancestrally polyploid than if they had no polyploid ancestry.

Breeding system shifts---changes in the collection of physiological and morphological traits that determine the likelihood that any two gametes unite---are remarkably common and crucially affect the distribution and amount of genetic variation in populations \citep{stebbins1974, barrett2013}.
In particular, self-incompatibility (SI) causes plants to reject their own pollen, and loss of self-incompatibility, yielding self-compatibility (SC), is one of the most replicated transitions in flowering plant evolution \citep{stebbins1974,igic_2008}. % Gordon: Really? I find this hard to believe... %B:  sigh.
Previous analyses reported higher rates of diversification for SI than for SC lineages in Solanaceae \citep{goldberg_2010}. 
Similarly, heterostylous SI lineages in Primulaceae seem to diversify faster \citep{devos2014}, as do outcrossing lineages in Onagraceae \citep{freyman_2018}.
Despite consistent inferences of their macroevolutionary role, breeding systems are clearly not the sole trait determining diversification rates.

Ploidy, for example, is widely thought to affect the propensity for self-fertilization \citep{stebbins1950}. 
%B: I did not find this confusing, but it worries me that Gordon did. Consequently, I've ejected this sentence, now focusing on SI/SC only.
%The evidence for correlated shifts in mating system, from outcrossing to selfing, following WGD, appears limited and sometimes contradictory \citep{barringer2007, barrett2008, husband2008}.
Polyploidy may facilitate the transition to selfing \citep{barringer2007, barrett2008, husband2008}, or it may directly cause the loss of SI \citep{stout1942, lewis1947}.
% Polyploidy is not a mere correlate of transition in breeding system, but may directly cause the loss of SI \citep{stout1942, lewis1947}, as well as facilitate the transition to selfing \citep{barringer2007, barrett2008, husband2008}.  E: reworded because no one would think the only thing P does is follow BS?
Doubled number of alleles in pollen is thought to effect disruption of the genetic mechanisms in gametophytic SI systems, which prevent self-fertilization \citep{entani1999, tsukamoto2005, kubo2010}. 
With exceedingly few exceptions \citep{hauck_2002,nunes_2006}, this generally precludes the existence of SI polyploids in the most widespread system of SI, which is regarded as ancestral in eudicots \citep{igic_2001,steinbachs_2002}.
This system, RNase-based SI, is expressed in all SI species of Solanaceae examined to date.
There is thus a strong correlation between polyploidy and SC in this family \citep{robertson_2011}.
% Gordon comments: Here again is confused about breeding systems in general vs. SI genetic systems.
%B: not confused here; some SI is broken down by ploidy. Therefore, there is a correlation with coefficient between 0 and 1, and not 1.0 
% Gordon comments here: First, it's a little confusing if you are talking about more SC or SI species more generally, or if you are talking specifically about genetic SI systems. I don't think the de Vos paper is dealing with SI systems. If your'e talking more generally about breeding system evolution there's a vast literature here, and I think  this is oversimplifying. 
% R- Boris what do you think is the best way to proceed here? Narrow it down to SI systems (deleting de Vos paper) or expanding it to talk about breeding systems in general and then saying that we will only focus on SI systems because of Solanaceae
%B: The apparent problem for Gordon is that devos2014 estimates *SSE parameters for heterostyly and homostyly, and not homomorphic SI. That's not a concern, I think, they are all "SI systems." And Dioecy etc. is not classically considered a "breeding system", but a "sexual system". That may be a part of the confusion. It's really not Gordon's problem, and we could do something to yet again re-define these terms? Emma?
% E: I vote for using terms correctly without providing a glossary---lead by example?

Given that changes in ploidy and breeding systems are causally related, and have profound affects on the fate of lineages, it seems particularly profitable to examine possible interactions in their macroevolutionary effects.
First, it may be of interest to establish their joint influence on speciation and extinction rates.
If the effects of ploidy and breeding system on lineage diversification are not additive, studying each trait separately may not reveal the effects of their combinations.
Second, the two traits may affect each other's state transitions.
For example, losses of SI in diploids could be caused by polyploidization, resulting in a one-step pathway to polyploid SC species. 
Alternatively, the path from SI diploids to SC polyploids may involve a transition to SC, without an increase in ploidy, and then a subsequent polyploidization. 
\citet{robertson_2011} found that the pathway from SI diploids to SC polyploids is dominated by the two-step pathway over long timescales, but generally proceeds in one step via polyploidization of SI species over short timescales.
It remains to be seen whether these results hold true when allowing for the influence of these traits on lineage diversification.

Two recent methodological developments enable disentangling the complex interactions between two or more traits and diversification.
First, \citet{fitzjohn_2012} extended the state-dependent diversification ``*SSE'' modeling framework \citep{maddison_2007} to allow estimating the effect of an arbitrary number of states (enabling multi-state, MuSSE, instead of binary state, BiSSE, speciation and extinction models).
This allows multiple states to represent trait combinations, particularly useful if one is interested in explicitly testing whether combinations of traits non-additively affect speciation or extinction. 
Second, \citet{beaulieu_2016}  proposed an extension of the *SSE framework by introducing a model with unspecified, or `hidden' traits (HiSSE model), which can affect the diversification process.
Because variation in speciation and extinction rates may not be due solely to the focal trait, HiSSE simultaneously enables hidden states to explain such variation, and it reduces the elevated false-positive error rates.
%B: This may be a good portion of Discussion--faults and qualities of methods?
%HiSSE models have effectively shown that some traits associated with diversification in BiSSE analyses are actually not associated with the diversification process, and instead a hidden state is linked to differences in the diversification rate.  
%Such results raise the question of the identity of the hidden state, or whether it is an approximation of some unknowable heterogeneity, as well as whether interactions between the known and hidden trait were modeled appropriately. 
%Third, one can simultaneously consider the effects of multiple known traits using MuSSE, especially in systems where multiple traits are suspected to influence diversification, and where interactions between those traits are well understood. 
%However, MuSSE might also suffer from hight type I errors if there are other processes shaping diversification other than the traits observed \citep{caetano_2018}
%Therefore, a fourth option exists: considering both the presence of multiple known traits  linked to the diversification process, but also, some unknown or hidden state changing speciation and extinction rates. 
A joint application of these advances allows simultaneous consideration of the effects of multiple known traits on the diversification process, along with the presence of other unobserved states that can affect speciation and extinction rates. 
%These four modeling strategies are proposed here in the context of studying complex interactions among traits, linked diversification.

Here, we examine the roles of ploidy and breeding system, as well as their interaction, on the diversification process in Solanaceae.
Using extensive trait data, and considering other possible sources of heterogeneity in the diversification process, we assess the influence of these traits on speciation and extinction rates.
First, we present models in which the effect of ploidy and breeding system are considered separately, and we compare the inferences to previously published results. 
Second, we add a hidden trait to each of those models and investigate whether the focal trait still explains differences in diversification.
Third, we model ploidy and breeding system jointly to assess their combined relationship to diversification, with or without an additional hidden trait.
With this joint trait model, we also quantify the relative contributions of the two pathways from SI diploids to SC polyploids.
For all of the ploidy models, we investigate the potential effects of including diploidization.
Our results highlight the importance of considering non-additive effects of traits on net diversification rates, especially when there are strong biologically driven correlations among them.

% R- I'm good with this intro. It is simple and reads well.