\section{Introduction}


% Outline, discussed by Boris and Emma:
%
% we are interested in the evolution of traits and their influence on lineage diversification
%     ref Barrett work 
% 		no diversification?, but trait shifts:
%		barrett_2008: "Because polyploidy affects the entire genome, it is perhaps not surprising that it influences many aspects of the phenotype, including the mating system. However, although it has long been recognized that the evolutionary transition from diploidy to polyploidy may result in correlated changes in mating patterns, the theoretical and empirical evidence is limited and often contradictory." (p.5)
%     any one study focuses on one/few traits
%     but any real organisms have many, so need to enlarge our view
%     mini "here, we..."
% we have tools to try to answer these questions
%     sse models
%     issues
%     but a way forward is to allow for multiple factors that could influence diversification
% ploidy
%     Stebbins
%     Mayrose, Soltis
%     recent paleo WGD, diploidization
% breeding system
%     directionality
%     diversification in Solanaceae
%     correlation with ploidy
%     pathways from SI-D to SC-P: direct or via SC-D
% here, we...
%     diversification: interaction between ploidy and breeding system
%     also: diploidization, pathways

Species accumulate around the tree of life at different rates.
The search for traits that explain these differences has been accelerated by dramatic increases in phylogenetic data and, despite some setbacks \citep{maddison_2015, rabosky_2015}, advances in analytical methods \citep{maddison_2007, fitzjohn_2009, goldberg_2012, beaulieu_2016}. % todo more refs: papers with big trees; quasse, bisse-ness, fisse, abc-quasse, maybe hoehna-vs-rabosky
In these studies, it is common to identify a single focal trait and investigate its association with rates of speciation and extinction. % (but see... ?) % refs: there must be some papers that musse on multiple traits? especially for geographic range?
This is problematic for two reasons.
On the biology side, the context in which traits occur can lead to complex interactions, perhaps with other traits that make up an organism or its environment.
On the methods side, excluding other traits that additionally influence diversification can lead to incorrect conclusions about the focal trait \citep{rabosky_2015, beaulieu_2016}.
One approach to this problem is to allow for a `hidden' or unspecified trait in the analysis \citep{beaulieu_2016}. % and an earlier beaulieu? applications of hisse?
This can be effective, but it leaves behind the question of whether the hidden trait corresponds to a real trait that should be sought, or whether it is an approximation of some unknowable heterogeneity, as well as whether interactions between the known and hidden trait were modeled appropriately.
Alternatively, we can simultaneously consider the effects of more than one known trait, especially in systems where multiple traits are suspected to influence diversification, and where interactions between those traits are well understood.
% B: We can then have a good prior, reliable data, and replicate the studies.  E: Agree. But when I tried to add it, I realized hidden models can also be hypothesis-generating, with the purported hidden trait identified in the next clade.
In this paper, we focus on two of the best studied traits in flowering plants, ploidy and breeding system, and investigate their individual and joint effects on diversification.

% E plan, merging the text from R and B:
%
% polyploidy is potentially important, widespread
% link with diversification; include diploidization hypothesis
%
% link with other traits -- correlated (geography), more directly connected (selfing, si)
%
% si
% si and ploidy
% pathways

Polyploidization multiplies the genomic content of cells, and it consequently has the potential to affect many or all traits and a great variety of evolutionary processes.
It is also a mutation that occurs commonly in plants, at it is widespread at both population and lineage scales.
The prevalence of variation in chromosome number, and especially ploidy, has been broadly considered a salient feature of flowering plants for nearly a century \citep{stebbins1938}. % B: instead of "ploidy" here, we may wish to use some stand-in for karyotype multiples, and stick to that language throughout. E: What word would that be? I was taking "ploidy" to mean "whatever multiple," but is that a terminological abuse?
The recent dramatic increase in the scale of available genome sequences uncovered ancient rounds of whole-genome duplications, and subsequent diploidization, across angiosperms \citep{lynch2000, vision2000}. % E: I am deferring to R's use of "diploidization" instead of "re-diploidization"
Nearly all lineages of flowering plants are thought to have undergone at least one or two rounds of polyploidization. % todo refs, jiao etc
%
The prevalence of polyploidy, its variation across clades, and its large effects on genotypes and phenotypes raises the hypothesis that ploidy plays an important role in shaping rates of speciation and extinction.
This has led to a debate: \citet{mayrose_2011, mayrose_2015} found a lower rate of lineage diversification for polyploids, while \citet{soltis_2014} argued that polyploidy is not an evolutionary dead end because whole genome duplications occurred at the base of many successful angiosperm clades, and \citet{landis_2018} reported higher diversification rates following some whole genome duplications. % and tank2015?
Our study revisits this question, additionally incorporating the process of diploidization.
% E:  Rosana, could you add a little about the importance of including diploidization?  todo
Furthermore, we consider the interaction between polyploidy and other traits.

From the very beginning of polyploidy research, the study of this cytogenetic property considered correlations with other traits \citep{stebbins1938}. 
Some of these associations are driven by indirect or ecological effects, such as the tendency for polyploids, and other traits, to be found at higher latitudes or more marginal habitats. % todo refs: husband etc
Other correlations may have a more direct, causal connection.
Among the many of changes associated with polyploidization, perhaps the most prominent is the association between polyploidy and propensity for self-fertilization \citep{stebbins1950, barrett1988}.
In some cases, the evidence for a correlated shift in mating system along with polyploidization appears limited and sometimes contradictory \citep{barringer2007, barrett2008, husband2008}.
In other cases, however, polyploidy is not only a suspected correlate of breeding systems but indeed a causal link \citep{stout1942, lewis1947}.
Doubled number of alleles in pollen is thought to effect disruption of the genetic mechanisms in gametophytic self-incompatibility systems, which prevent self-fertilization \citep{entani1999, tsukamoto2005, kubo2010}. 
In clades with these systems, it is thus natural to consider the simultanous macroevolution of ploidy and breeding system.

% E note to self: edit this next...
Breeding system shifts---changes in the collection of physiological and morphological traits that determine the likelihood that any two gametes unite---are remarkably common and affect the distribution and amount of genetic variation in populations \citep{stebbins1974,barrett2013}.
They are by themselves thought to have a broad effect on other traits. 
The frequent transition from self-incompatibility (SI) to self-compatibility (SC) is strongly associated with net diversification rate changes \citep{goldberg_2010,devos2014}.
Given that changes in ploidy and breeding systems may be causally related and have profound affects on the fate of lineages, it seems particularly profitable to examine whether and how their macroevolutionary effects interact.
Moreover, increases in ploidy are known to directly affect breeding system states and present a rare opportunity for the study of paths and path-dependence on trait evolution and diversification.
For example, we could ask if the ordinary mutation (and fixation) causes more losses of SI than polyploidization, whether more SC polyploids are generated directly, after polyploidization of diploid SI lineages, or through a two-step pathway, involving loss of SC and subsequent polyploidization of a SC lineage.
And, are diversification rates of polyploid SC lineages different if one or other path was taken? \citet{charlesworth1985} 
%
%trans-polymorphism: We also have this great data for SI loci and the transition could be even more common, providing power to resolve the great diversification questions
%
% set up pathway dependence from SI-D to SC-P: direct or via SC-D


Here, we [...]
% Someone else should do this, once we have the full data.
% diversification: interaction between ploidy and breeding system
% 	examine individual effects
% 	examine joint effects
% 	with and without HMMs
% 	allow for re-diploidization or not
% also: diploidization, pathways
% 	examine pathways from SI-D to SC-P: direct or via SC-D


%The second question (talk here about breeding system and polyploidy, and how this could be different across clades and this is one of the reasons why we focused on Solanaceae).
%-Diversification and breeding systems\\
%Goldberg and Igic 2012, different  perspectives\\
