\section{Introduction}


% Outline, discussed by Boris and Emma:
%
% we are interested in the evolution of traits and their influence on lineage diversification
%     ref Barrett work 
% 		no diversification?, but trait shifts:
%		barrett_2008: "Because polyploidy affects the entire genome, it is perhaps not surprising that it influences many aspects of the phenotype, including the mating system. However, although it has long been recognized that the evolutionary transition from diploidy to polyploidy may result in correlated changes in mating patterns, the theoretical and empirical evidence is limited and often contradictory." (p.5)
%
%     any one study focuses on one/few traits
%     but any real organisms have many, so need to enlarge our view
%     mini "here, we..."
% we have tools to try to answer these questions
%     sse models
%     issues
%     but a way forward is to allow for multiple factors that could influence diversification
% ploidy
%     Stebbins
%     Mayrose, Soltis
%     recent paleo WGD, diploidization
% breeding system
%     directionality
%     diversification in Solanaceae
%     correlation with ploidy
%     pathways from SI-D to SC-P: direct or via SC-D
% here, we...
%     diversification: interaction between ploidy and breeding system
%     also: diploidization, pathways

%Studying diversification linked to trait evolution\newline
%Here I need help (perhaps will be the last line I write). In these context, botanist have been asking how polyploidy shapes patterns of diversification. The prevalence of polyploidy and its detection across multiple and highly diverse clades of angiosperms inevitably lead to the hypothesis of the importance of polyploidy in the speciation and extinction patterns observed across the flowering plant phylogeny.  At the same time, a similar question has been asked in the breeding system world, where self-compatibility has evolved multiple times in flowering plants. However, the influence in diversification using both polyploidy and self-compatibility information has not been studied simultaneously.

%Why polyploidy and self-compatibility in particular makes for an  interesting study in the context of diversification\newline In the polyploidy world, an important debate regarding the diversification of angiosperms has been ongoing since the publication of \citet{mayrose_2011}. The authors discovered that using the latest diversification model linked to two states diploid and polyploid, the net diversification of polyploids was much slower than the net diversification rate than polyploids. This result was surprising and an sparked a discussion about the long-term evolutionary consequences of polyploidy. \citet{soltis_2014} questioned if polyploidy should be regarded as an evolutionary dead-end since the net diversification of polyploids was negative, despite overwhelming evidence about the incidence of polyploidy  especially at the root of highly diverse angiosperm claims (ref here). A year later, diversification models were re-tested and corrected, and still found the same pattern \citep{mayrose_2015}, to the disappointment of plenty of botanists there was no denying of the weak trend of diversification that polyploids left behind. The most recent study by \citet{landis_2018} that used not only the presence of polyploidy in the tips but also the number of whole genome duplications in a taxon lineage found that.... \newline  Meanwhile, studies focusing on diversification patterns and  breeding system have consistently found that self-incompatible plants often have higher net diversification rates compare to their self-compatible counterparts (Emma and Boris' papers here, what about papers that are not solanaceae?) 

%- Why other traits need to be considered as well\newline
%- What other models and studies have done in the past\newline
%- What is lacking from past approaches? \newline
%There are two key questions that at the time of the polyploidy debate were difficult to ask. The first is if the models used to measure the diversification of polyploids were correct, and the second question is if the models have potentially included more evidence and potential traits that are not polyploidy or other lines of evidence that could be driving the patterns. At the time, in a different context \citet{beaulieu_2016} were finding an alternative solution to the first question, coming up with a new model that could represent the broad heterogeneity of the diversification process and parse out the signal between the trait of interest and the noise in diversification. Their model, the hidden state speciation and extinction model is a key component to detect whether polyploidy our something else unknown but related to it is driving the speciation and extinction patterns that we see in angiosperms. \newline

Species accumulate around the tree of life at different rates.
The search for traits that explain these differences has been accelerated by dramatic increases in phylogenetic data and, despite some setbacks \citep{maddison_2015, rabosky_2015}, advances in analytical methods \citep{maddison_2007, fitzjohn_2009, goldberg_2012, beaulieu_2016}. % todo more refs: papers with big trees; quasse, bisse-ness, fisse, abc-quasse, maybe hoehna-vs-rabosky
In these studies, it is common to identify a single focal trait and investigate its association with rates of speciation and extinction. % (but see... ?) % refs: there must be some papers that musse on multiple traits? especially for geographic range?
This is problematic for two reasons.
On the biology side, the context in which traits occur can lead to complex interactions, perhaps with other traits that make up an organism or its environment.
On the methods side, excluding other traits that additionally influence diversification can lead to incorrect conclusions about the focal trait \citep{rabosky_2015, beaulieu_2016}.
One approach to this problem is to allow for a `hidden' or unspecified trait in the analysis \citep{beaulieu_2016}. % and an earlier beaulieu? applications of hisse?
This can be effective, but it leaves behind the question of whether the hidden trait corresponds to a real trait that should be sought, or whether it is an approximation of some unknowable heterogeneity, as well as whether interactions between the known and hidden trait were modeled appropriately.
Alternatively, we can simultaneously consider the effects of more than one known trait, especially in systems where multiple traits are suspected to influence diversification, and where interactions between those traits are well understood.
% B: We can then have a good prior, reliable data, and replicate the studies.  E: Agree. But when I tried to add it, I realized hidden models can also be hypothesis-generating, with the purported hidden trait identified in the next clade.
In this paper, we focus on two of the best studied traits in flowering plants, ploidy and breeding system, and investigate their individual and joint effects on diversification.

The prevalence of variation in chromosome number, and especially ploidy, has been broadly considered a salient feature of flowering plants for nearly a century. %\citep{stebbins1938}; instead of "ploidy" here, we may wish to use some stand-in for karyotype multiples, and stick to that language throughout.
The recent dramatic increase in the scale of available genome sequences uncovered ancient rounds of whole-genome duplications, and subsequent re-diploidization, across angiosperms \citep{lynch2000,vision2000}. 
Nearly all lineages of flowering plants are thought to have undergone at least one or two rounds of polyploidization. %cite
It is therefore surprising that there is little agreement regarding the evolutionary consequences of polyploidy, especially how it affects diversification rates. %species richness; cite: mayrose2011, soltis2014, mayrose2015, tank2015, landis2018?

Polyploidization alters the genomic content of all cells, and consequently has the potential to affect a wide variety, if not all, traits. 
From the very beginning, the study of this cytogenetic property considered correlations with other traits \citep{stebbins1938}. 
Among the many of changes associated with polyploidization, perhaps the most prominent is the association between polyploidy and propensity for self-fertilization \citep{stebbins1950, barrett1988}.
Polyploidy is not only a suspected correlate of breeding systems but a causal link \citep{stout1942, lewis1947}.
Doubled number of alleles in pollen is strongly suspected to effect disruption of the genetic mechanisms in gametophytic self-incompatibility systems \citep{entani1999, tsukamoto2005, kubo2010}. 
Polyploidy is also thought to be associated with changes in the rate of self-fertilization, although the evidence for this correlated shift in mating system appears limited and sometimes contradictory \citep{barringer2007, barrett2008, husband2008}.

Breeding system shifts---changes in the collection of physiological and morphological traits that determine the likelihood that any two gametes unite---are remarkably common and affect the distribution and amount of genetic variation in populations \citep{stebbins1974,barrett2013}.
They are by themselves thought to have a broad effect on other traits. 
The frequent transition from self-incompatibility (SI) to self-compatibility (SC) is strongly associated with net diversification rate changes \citep{goldberg_2010,devos2014}.
Given that changes in ploidy and breeding systems may be causally related and have  profound affects on the fate of lineages, it seems particularly profitable to examine whether and how their macroevolutionary effects interact.
Moreover,  increases in ploidy are known to directly affect breeding system states and present a rare opportunity for the study of paths and path-dependence on trait evolution and diversification.
For example, we could ask if the ordinary mutation (and fixation) causes more losses of SI than polyploidization, whether more SC polyploids are generated directly, after polyploidization of diploid SI lineages, or through a two-step pathway, involving loss of SC and subsequent polyploidization of a SC lineage.
And, are diversification rates of polyploid SC lineages different if one or other path was taken? \citet{charlesworth1985} 
%
%trans-polymorphism: We also have this great data for SI loci and the transition could be even more common, providing power to resolve the great diversification questions
%
% set up pathway dependence from SI-D to SC-P: direct or via SC-D


Here, we [...]
% Someone else should do this, once we have the full data.
% diversification: interaction between ploidy and breeding system
% 	examine individual effects
% 	examine joint effects
% 	with and without HMMs
% 	allow for re-diploidization or not
% also: diploidization, pathways
% 	examine pathways from SI-D to SC-P: direct or via SC-D


%The second question (talk here about breeding system and polyploidy, and how this could be different across clades and this is one of the reasons why we focused on Solanaceae).
%-Diversification and breeding systems\\
%Goldberg and Igic 2012, different  perspectives\\
