\begin{quote}
\em{``Among life history traits, reproductive characters that determine mating system patterns are perhaps the most influential in governing macroevolution." 
}
\\
\hspace*{\fill}\rm{ Barrett et al. (1996)}
\end{quote}

\section{Introduction}

Species accumulate across the tree of life at different rates. 
A possible explanation for this phenomenon is that species possess various traits or character states that differentially affect rates of diversification. 
Dramatic increases in the available phylogenetic and phenotypic data, as well as methodological advances, have greatly accelerated the search for traits that influence diversification traits.
Nevertheless, identifying focal traits associated with rates of speciation and extinction remains a difficult problem \citep[\eg][]{maddison_2015, rabosky_2015, moore_2016, fitzjohn_2009, goldberg_2012, beaulieu_2016, rabosky_2017}. 
Problems arise at least in part because speciation and extinction may not depend on a single trait, and the context in which traits occur can lead to complex interactions resulting in a heterogenous speciation and extinction rates \citep{beaulieu_2016, caetano_2018, herrera_2018}.
Consequently, examining the association of only one trait with diversification patterns can be misleading. 

Polyploidization is a remarkably common mutation in plants \citep{husband_2013, zenilferguson_2017}.
The widespread prevalence of variation in ploidy has long been considered a salient feature of flowering plant lineages \citep{stebbins1938}. 
Increase in ploidy can alter many traits and impact a variety of evolutionary \citep{ramsey_2002} and ecological processes \citep{sessa_2019}.
%
Polyploids are thought to have lower diversification rate than diploids \citep{mayrose_2011, mayrose_2015}. 
Recent studies, however, find common and numerous paleo-polyploidizations, including some preceding the emergence of highly diverse plant clades \citep{soltis_2014, landis_2018}, suggesting that whole genome duplications (WGDs) have played an important macroevolutionary role driving innovation and diversification in plants. 
Evidence of paleo-polyploidy within the genomes of diploid extant plants also implies pervasiveness of diploidization, the return of polyploids to the diploid state, throughout the angiosperm phylogeny \citep{soltis_2015, dodsworth_2015}.
This could affect the interpretation of different net diversification rates for diploids and polyploids, especially if diploids have a relatively higher net diversification rate if they were ancestrally polyploid than if they had no polyploid ancestry.

Breeding system shifts---changes in the collection of physiological and morphological traits that determine the likelihood that any two gametes unite---are remarkably common and crucially affect the distribution and amount of genetic variation in populations \citep{stebbins1974, barrett2013}.
In particular, self-incompatibility (SI) causes plants to reject their own pollen, and loss of self-incompatibility, yielding self-compatibility (SC), is one of the most replicated transitions in flowering plant evolution \citep{stebbins1974,igic_2008}.
Previous analyses reported higher rates of diversification for SI than for SC lineages in Solanaceae \citep{goldberg_2010}. 
Similarly, heterostylous SI lineages in Primulaceae seem to diversify faster \citep{devos2014}, as do outcrossing lineages in Onagraceae \citep{freyman_2018}.
Despite consistent inferences of their macroevolutionary role, breeding systems are clearly not the sole trait determining diversification rates, and may exhibit complex interactions with other traits.

Ploidy, for instance, is widely thought to affect the propensity for self-fertilization \citep{stebbins1950}. 
Whole genome duplication may help facilitate the loss of SI and the transition to selfing \citep{barringer2007, barrett2008, husband2008}, or it may directly cause the loss of SI \citep{stout1942, lewis1947}.
Doubled number of alleles in pollen is thought to effect disruption of the genetic mechanisms in gametophytic SI systems, which prevent self-fertilization \citep{entani1999, tsukamoto2005, kubo2010}. 
With exceedingly few exceptions \citep{hauck_2002,nunes_2006}, this generally precludes the existence of SI polyploids in the most widespread system of SI, which is regarded as ancestral in eudicots \citep{igic_2001,steinbachs_2002}.
This system, RNase-based SI, is expressed in all SI species of Solanaceae examined to date.
There is consequently a strong correlation between polyploidy and SC in this family \citep{robertson_2011}.

%B: removed First, ... Second, ... because we had three consecutive counting paragraphs.
Given that changes in ploidy and breeding systems are causally related, and may have profound effects on the fate of lineages, it seems particularly profitable to examine possible interactions in their macroevolutionary effects.
If the effects of ploidy and breeding system on lineage diversification are not additive, studying each trait separately may not reveal the effects of their combinations.
As a result, it may be of interest to establish their joint influence on speciation and extinction rates.
The two traits may also affect each other's state transitions.
For example, losses of SI in diploids could be caused by polyploidization, resulting in a one-step pathway to polyploid SC species. 
Alternatively, the path from SI diploids to SC polyploids may involve a transition to SC, without an increase in ploidy, and then a subsequent polyploidization. 
\citet{robertson_2011} found that the pathway from SI diploids to SC polyploids is dominated by the two-step pathway over long timescales, but generally proceeds in one step via polyploidization of SI species over short timescales.
It remains to be seen whether these results hold true when allowing for the influence of these traits on lineage diversification.

%B: removed First, ... Second, ... because we had three consecutive counting paragraphs.
Two recent methodological developments enable disentangling the complex interactions between two or more traits and diversification.
\citet{fitzjohn_2012} extended the state-dependent diversification ``*SSE'' modeling framework \citep{maddison_2007} to allow estimating the effect of an arbitrary number of states (enabling multi-state, MuSSE, instead of binary state, BiSSE, speciation and extinction models).
This allows multiple states to represent trait combinations, particularly useful if one is interested in explicitly testing whether combinations of traits non-additively affect speciation or extinction. 
Soon thereafter, \citet{beaulieu_2016} proposed an extension of the *SSE framework by introducing a model with unspecified, or `hidden' traits (HiSSE model), which can affect the diversification process.
Because variation in speciation and extinction rates may not be solely due to the focal trait, HiSSE simultaneously enables hidden states to explain such variation, and it reduces the elevated false-positive error rates.
A joint application of these advances allows simultaneous consideration of the effects of multiple known traits on the diversification process, along with the presence of other unobserved states that can affect speciation and extinction rates. 

Here, we examine the roles of ploidy and breeding system, as well as their interaction, on the diversification process in Solanaceae.
Using extensive trait data, and considering other possible sources of heterogeneity in the diversification process, we assess the influence of these traits on speciation and extinction rates.
First, we present models in which the effect of ploidy and breeding system are considered separately, and we compare the inferences to previously published results. 
Second, we add a hidden trait to each of those models and investigate whether the focal trait still explains differences in diversification.
Third, we model ploidy and breeding system jointly to assess their combined relationship to diversification, with or without an additional hidden trait.
With this joint trait model, we also quantify the relative contributions of the two pathways from SI diploids to SC polyploids.
For all of the models involving ploidy, we investigate the potential effects of including diploidization.
Our results highlight the importance of considering non-additive effects of traits on net diversification rates, especially when there are strong biologically driven correlations among them.