\section{Methods}

\subsection{Data}

Chromosome number data were obtained for all Solanaceae taxa in the the Chromosome Counts Database \citep[CCDB;][]{rice_2015}, and the ca.\ 14,000 records were cleaned semi-automatically using the CCDBcurator R package \citep{zenilferguson_2017}.
This large dataset includes the compilation of Solanaceae ploidy states from \citep{robertson_2011}.
Species were coded as either diploid (D) or polyploid (P).
For the majority of species, ploidy was assigned according to information from the original publications and the Kew Royal Botanic Gardens C-value DNA resource \citep{bennett_2005}.
For taxa without ploidy information but with information about chromosome number, we assigned ploidy based on the multiplicity of chromosomes within the genus.
For example, \textit{Solanum betaceum} did not include information about ploidy level but it has 24 chromosomes, so because $x=12$ is the base chromosome number of the \textit{Solanum} genus \citep{olmstead_2007}, we assigned \textit{S.~betaceum} as diploid. 
Species with more than one ploidy level were assigned the smallest most frequent ploidy level recorded. % E: Does this apply to the diploid-polyploid levels?  If so, we need to explain how it is also reconciled against the SI/SC value.  Or is this sentence just about multiple levels of polyploidy, which wouldn't need to be mentioned here?

Breeding system was scored as self-incompatible (SI) or self-compatible (SC) based on results hand-curated from the literature (as in \citealt{igic_2006, goldberg_2010, robertson_2011, goldberg_2012}).
Most species could unambiguously be coded as either SI or SC \citep{raduski_2012}.
Following previous work, we coded as SI any species with functional SI systems, even if SC or dioecy was also reported.
Dioecious species without functional SI were coded as SC.

To those existing data sets, we added some additional records for chromosome number and breeding system.
The Supplementary Information contains citations for the numerous original sources for all of the data. % todo: add number of refs when we have it
Synonymy followed Solanaceae Source \citep{solsource}.
Hybrids and cultivars were excluded because ploidy and breeding system are likely to be altered by domestication.
As in \citet{robertson_2011}, we examined closely the few species for which the merged ploidy and breeding system data indicated the presence of SI polyploids.
Although SI populations frequently contain some SC individuals and diploid populations frequently contain some polyploid individuals, in no case did we find individuals that were both SI and polyploid.
Because of this empirical observation and the functional explanation for whole genome duplication disabling gametophytic self-incompatibility \citep[reviewed in][]{ramsey_1998,stone_2002}, we consider only three observed character states: SI-D, SC-D, and SC-P.

% E: We should probably include some more details about how many polymorphic species and how the particularly tricky ones were dealt with.

Matching our character state data to the largest time-calibrated phylogeny of Solanaceae \citep{sarkinen_2013} yielded 595 species with ploidy and/or breeding system information on the tree.
Binary or three-state classification of ploidy and breeding system for the 595 taxa is summarized in \cref{table:stateclassifications}. % E: I think you're still updating the state coding?
We retained all of these species in each of the analyses below, because pruning away tips lacking breeding system in the ploidy-only analyses (and vice versa) would discard data that could inform the diversification models.
% E: Though someone might ask why tips with no state data were pruned.  Analyses took long enough on even this-size tree.

\subsection{Models}

% E: There are so many models, I think we need to try hard to help the reader keep them straight.
% I would avoid naming any states with numbers, except maybe within the stateclassification table and its caption.
% I also still find bisse/hisse/muhisse terminology confusing, I think because it both describes a general class of model and applies to a specific analysis here, and because bisse and hisse are run separately on ploidy breeding system.
% How about a model naming scheme like this?
%   D/P
%   D/P-A/B
%   I/C
%   I/C-A/B
%   ID/CD/CP
%   ID/CD/CP-A/B
% Or something else more concise and descriptive?

We first defined a binary state speciation and extinction model (BiSSE, \citet{maddison_2007})  for polyploidy evolution where taxa was classified as diploid (D=0) and polyploid (P=1) \cref{table:stateclassifications}.
In a Bayesian framework, we obtained posterior probability distributions of speciation rates ($\lambda_D$, $\lambda_P$), extinction rates ($\mu_D$, $\mu_P$) , net diversification rates ($r_D=\lambda_D-\mu_D, r_P=\lambda_P-\mu_P$), and  relative extinction rates ($\nu_D=\frac{\mu_D}{\lambda_D}, \nu_D=\frac{\mu_D}{\lambda_D}$) just as previously explored in \citet{mayrose_2011}.
The cladogenetic changes explored in BiSSE were polyploidization rate are represented by parameter $\rho$ and by diploidization parameter rate $\delta$.
% E: cladogenetic?

As a second step, we fitted a hidden state speciation and extinction model (HiSSE, \citet{beaulieu_2016}) to evaluate whether the differences in diversification rates were found due to a hidden trait associated to polyploidy.
As discussed by \citet{beaulieu_2016} BiSSE-like models suffer from a large type I error because they fail to account as part of the null hypothesis heterogeneous diversification rate changes  that do not depend on the trait of interest.
By including a hidden trait in a state speciation and extinction  model,  HiSSE-like models address the heterogeneous background diversification changes while also parsing out the possible signal of diversification due to the trait of interest. 
Therefore, HiSSE model for polyploidy linked to diversification has four states: diploid and polyploid subdivided by a binary hidden trait with states A and B for which we estimated the posterior probability distributions of speciation rates ($\lambda_{D_A},\lambda_{D_B}, \lambda_{P_A},\lambda_{P_B}$), extinction rates ($\mu_{D_A},\mu_{D_B}, \mu_{P_A},\mu_{P_B}$),  net diversification rates ($r_{D_A},r_{D_B},r_{P_A},r_{P_B}$), and relative extinction($\nu_{D_A},\nu_{D_B}, \nu_{PA},\nu_{P_B}$).
The cladogenetic changes assumed by the fitted model are polyploidization rate $\rho$, diploidization rate $\rho$, and we assumed that the changes between every hidden state are symmetrical with rate $\alpha$.

For breeding system both BiSSE and a HiSSE models were fitted independently by coding data as self-compatible (SC=0) or self-incompatible (SI=1) \cref{table:stateclassifications}.
The BiSSE model fitted to breeding system data was done with the goal of investigating if the effect of self-incompatibility in diversification was similar to the pattern found by \citet{goldberg_2012}.
However, the HiSSE model on breeding system parses out if the effect of breeding system in diversification is significant  by minimizing type I error as mentioned above.
For these models we assumed that self-compatibility is irreversible as discussed by \citet{igic_2013}, so we defined the transition rate $q_{IC}$ as the parameter for changes from self-incompatible to self-compatible state.
% E: explain irreversibility assumption; fix ancestry instead?

Next, we proposed a multivariate speciation and extinction model (MuSSE, \citet{fitzjohn_2012}) to investigate the link amongst diversification, breeding system, and polyploidy simultaneously.
The MuSSE model is defined using three states self-compatible diploids (SC-D=0), polyploids that are always self-compatible (ref for this??) (P=1), and self-incompatible diploids (SI-D=2, see table \cref{table:stateclassifications}) and contains ten parameters, six defining diversification ($\lambda_{SD}, \mu_{SD},\lambda_{P},\mu_{P}, \lambda_{ID},\mu_{ID}$) and the other four key cladogenetic changes that are: polyploidization of self-compatible diploids $\rho_{SD}$, diploidization $\delta$, polyploidization of self-incompatible diploids $\rho_{ID}$, and self-incompatible to self-compatible rate $q_{IC}$.
% E: for SC, use C instead of S
% E: lack of SI-P is now mentioned in Data and should be explained in Intro

Since the null hypothesis of the MuSSE model is that the diversification is equal and constant for all three states defined, it is possible that MuSSE can also suffer from large type I errors.
In order to account for heterogeneity of the diversification rates and parse the signal in diversification coming from breeding system and polyploidy we extended MuSSE model to account for a hidden state, as \citet{beaulieu_2016} did in the bivariate case.
The model we proposed is a multivariate and hidden states speciation and extinction stochastic process (MuHiSSE) that allowed us to account simultaneously for the diversification rates linked to breeding system and polyploidy but also the presence of some more heterogeneity in the process.
This model is analogous to the GeoHiSSE model proposed by Caetano et al. (2018) in a biogeographical context.
The full MuHiSSE model has 26 parameters, however, our goal was to look for diversification rate differences so we fitted a simplified version of 17 parameters by fixing the rates amongst hidden states to be equal with parameter $\alpha$ and the transition rates amongst breeding system and polyploidy as defined in the MuSSE model  ($\rho_{SD}$,  $\delta$,  $\rho_{ID}$,  $q_{IC}$) despite the hidden state.
Using MuHiSSE we estimated twelve speciation and extinction rates ($\lambda_{SD_A}, \mu_{SD_A},\lambda_{P_A},\mu_{P_A}, \lambda_{ID_A},\mu_{ID_B},\lambda_{SD_B}, \mu_{SD_B},\lambda_{P_B},\mu_{P_B}, \lambda_{ID_B},\mu_{ID_B}$), and the net diversification and relative extinction rates associated with them.
% E: caetano ref; and huang

All the models were performed using RevBayes \citep{hoehna_2016} software that performs Bayesian inference via MCMC using Metropolis-Hastings algorithm.
A correction for sampling bias was done in all models by assuming that Solanaceae family has approximately 3,000 species ($s=595/3000$) as the Solanaceae Source project indicates \citep{solsource}.
In all models, speciation and extinction parameters used log-normal prior distributions that with mean the expected net diversification rate $ ( \frac{(number of taxa)/2}{root age})$ and standard deviation $0.5$.
Prior distributions for parameters defining cladogenetic changes were gamma distributed with parameters $k=0.5$ and $\theta=1$. 
MCMC was performed for 96 hours in the cluster at Minnesota Supercomputing Instute which allowed for 5,000 generation of burn-in and a minimum of 200,000 generations of MCMC for each of the 6 models.
Convergence of the MCMC was tested using R package coda (ref) and software Tracer (ref) to assess convergence and mixing (see supplementary information).


\begin{table}
\begin{tabular}{@{}llccc@{}} \toprule
\multicolumn{4}{r}{Models} \\ \cmidrule(r){3-5}
Type & Total of & BiSSE/HiSSE & BiSSE/HiSSE & MuSSE/MuHiSSE\\ 
 &Taxa &  Polyploidy & Breeding System &  \\ \midrule
Diploid Self Compatible & 152 & 0 &  0 & 0 \\
Diploid Self Incompatible& 97 & 0  & 1 & 2\\
Diploid with unknown breeding system & 219 & 0 & (0,1) & (0,2) \\
Polyploid & 81 & 1& 0 & 1 \\
Unknow ploidy and self compatible& 34 & (0,1)& 0 & (0,1) \\ 
Unknown ploidy and self incompatible & 12 & 0 & 1 & 2 \\ \bottomrule
\end{tabular}
\caption{Binary and three state classifications for 595 taxa with ploidy and/or breeding system data. The number of taxa in the t he sample was maximize by including tips with only ploidy or only breeding system and assigned them as uncertain in the unknown character.}
% E: but P are not SI
\label{table:stateclassifications}
\end{table}
