\section{Methods}

\subsection{Data}

%B: IMO: The data subsection would be better organized with parts, (1) ploidy, (2) breeding system, (3) tree, (4) models, (5) model selection and statistical inference.
% E: I changed the subsectioning, but to a different arrangement: (1) Data, (2) Models, (3) Statistical inference.  With subsubsections as necessary.

Chromosome number data were obtained for all Solanaceae taxa in the Chromosome Counts Database \citep[CCDB;][]{rice_2015}, and the ca.~14,000 records were curated semi-automatically using the CCDBcurator R package \citep{rivero_2019}.
CCDB contains records from original sources that have multiple complex symbol patterns denoting multivalence, or irregularites of chromosome counts.
After a first round of automatic cleaning, we examined results by hand and corrected records as necessary.
Our hand-curated records were also contrasted against the ploidy dataset from \citet{robertson_2011}, original references therein, and against ploidy data in the C-value DNA dataset from \citet{bennett_2005}.
By comparing three different sources of information, we were able to code taxa as diploid, $D$, or polyploid, $P$.
For the majority of species, ploidy was assigned according to information from the original publications included in the  C-value DNA dataset \citep{bennett_2005}.
For taxa without ploidy information but with information about chromosome number, we assigned ploidy based on the multiplicity of chromosomes within the genus/family, or based on SI/SC classification.
For example, \textit{Solanum betaceum} did not have information about ploidy level but it has 24 chromosomes, so since $x=12$ is the base chromosome number of the genus \textit{Solanum} \citep{olmstead_2007}, we assigned \textit{S.~betaceum} as diploid. 
Additionally, because of the absence of SI polyploids (explained above and below), species known to be SI could be scored as diploid.
Species with more than one ploidy level were assigned the most frequent ploidy level recorded or the smallest ploidy in case of frequency ties.

Breeding system was scored as self-incompatible, $I$, or self-compatible, $C$, based on results curated from the literature and original experimental crosses \citep[as compiled in][]{igic_2006, goldberg_2010, robertson_2011, goldberg_2012}.
Most species could unambiguously be coded as either $I$ or $C$ \citep{raduski_2012}.
Following previous work, we coded any species with a functional SI system as $I$, even if SC or dioecy was also reported.
Dioecious species without functional SI were coded as $C$.
% E: I am trying to stick with I/C when talking about the state values, but using SI/SC when talking about the meaning of the trait (because this what people use normally). 
%R- Okay- I was getting confused as well thanks for clarifying

Resolution of taxonomic synonymy followed Solanaceae Source \citep{solsource}. 
Hybrids and cultivars were excluded because ploidy and breeding system can be affected by artificial selection during domestication.
Following the reasoning outlined in \citet{robertson_2011}, we closely examined the few species for which the merged ploidy and breeding system data indicated the presence of self-incompatible polyploids.
%B: Modified this section for post-review response:
Although SI populations frequently contain some SC individuals, and diploid populations frequently contain some polyploid individuals, in no case did we find convincing data for a naturally occurring SI polyploid population  \citep[discussed in][]{robertson_2011}.
%B rm: The single instance of an SI polyploid individual appears to be an allopentaploid hybrid of \textit{Solanum oplocense} Hawkes x \textit{Solanum gourlayii} Hawkes, reported by \citet{camadro_1981}.
% Under exceedingly rare circumstances, it is possible for polyploids containing multiple copies of S-loci to remain SI, so long as they express a single allele at the S-locus \citep[discussed in][]{robertson_2011}.
Because of the resulting absence of polyploid SI populations, as well as the functional explanation for polyploidy disabling gametophytic SI systems with non-self recognition (see the Introduction), we consider only three observed character states: self-incompatible diploids $(ID)$, self-compatible diploids $(CD)$, and self-compatible polyploids $(CP)$.

Matching our character state data to the largest time-calibrated phylogeny of Solanaceae \citep{sarkinen_2013} yielded 651 species with ploidy and/or breeding system information on the tree.
% FIXME: Rosana will update numbers around here TODO
Of these, 290 had information for both states.
The number of species in each combination of states is summarized in \cref{figure:stateclassifications}A.
We retained all 651 species in each of the analyses below because pruning away tips lacking breeding system in the ploidy-only analyses (and vice versa) would discard data that could inform the diversification models.
A total of 405 taxa without any information about breeding system or ploidy were excluded.
%Tips without trait data might not improve point estimation of diversification parameters while increasing uncertainty in the estimates of trait linked diversification
% R-This is very important but I have removed it because I think it is a full on paper by its own
%Including this many more species would have prohibitively slowed our analyses, especially those implementing the most complex models.

The Supplementary Information contains citations for the numerous original data sources. % FIXME (and say here how many refs?)
The Dryad archive contains the data and tree files used for analyses. % FIXME
% \url{https://github.com/roszenil/solploidy}  E: we should use dryad instead, because github repos are not guaranteed to last

\subsection{Models}

To test our hypotheses about lineage diversification and trait macroevolution, we fit a large suite of phylogenetic models.
All models are from the SSE framework \citep[BiSSE, MuSSE, HiSSE;][]{maddison_2007, fitzjohn_2012, beaulieu_2016}.
This style of model has parameters that describe per-lineage rates of speciation and extinction, specific to each character state (denoted $\lambda$ and $\mu$, respectively, with subscripts to indicate the state), along with rates of transitions between states (denoted $\rho$ for polyploidization, $\delta$ for diploidization, and $q_{IC}$ for loss of self-incompatibility).

The full set of models and all their rate parameters are detailed in \cref{fig:allmodels}.
Here we summarize how each type of model allows us to assess whether diversification is best explained by ploidy, breeding system, their combination, or some unknown factor.

\subsubsection{Ploidy and diversification}

In order to investigate the association between ploidy level and diversification, we first defined a model in which taxa were classified as diploid $(D)$ or polyploid $(P)$ (M1).
This is the same model \citet{mayrose_2011} used for this question across an array of plant genera.
% R- Itay wanted to hightlight that is the same model. Definitely a BiSSE. He did the ClaSSE later in the 2014 response.
Although this model can be a powerful means to distinguish state-specific speciation and extinction from trait changes, it is prone to incorrectly reporting that a trait is associated with diversification differences \citep{maddison_2015, rabosky_2015}.
We therefore define several models that incorporate additional forms of diversification rate heterogeneity.

Our second ploidy model includes a binary hidden trait that subdivides each observed state (M2).
In this CID model \citep{beaulieu_2016}, the hidden trait can affect diversification but the observed trait does not.
Comparing M1 and M2 allows us to test whether diversification rate heterogeneity is better explained by ploidy or by some unknown factor.

We additionally fit three models in which both ploidy and a hidden trait could influence diversification (M3--M5).
These models differ in whether transitions between the hidden states are symmetric or asymmetric, and whether the polyploidization rate depends on the hidden state.
Comparing M1--M5 allows us to test whether ploidy is associated with diversification differences on top of differences explained by an unknown factor.

We further fit the analogues of these five models but including a rate parameter for transitions from polyploid to diploid (M6--M10).
This allows us to assess whether our conclusions about ploidy and diversification are robust to the possibility of diploidization.

\subsubsection{Breeding system and diversification}

Our first five breeding system models follow the same logic as the first ploidy models.
We begin with each species classified as self-incompatible $(I)$ or self-compatible $(C)$ (M11).
This is the same as the analysis of \citet{goldberg_2010} but with an updated phylogeny \citep{sarkinen_2013}.

We then add models to allow diversification to be influenced by only a hidden trait (M12), or by both breeding system and a hidden trait (M13--M15, with varying degrees of complexity in the hidden trait transitions).
Similar models were used by \citet{freyman_2017} used to study diversification in Onagraceae.

Self-incompatibility is homologous in all Solanaceae species in which S-alleles have been cloned and controlled crosses performed.
All species sampled to date possess a non-self recognition, RNase-based gametophytic self-incompatibility \citep[shared even with other euasterid families;][]{ramanauskas_2017}.
Furthermore, species that are distantly related within this family carry closely-related alleles, with deep trans-specific polymorphism at the locus that controls the SI response \citep{ioerger_1990, igic_2006}.
This represents very strong evidence that the $I$ state is ancestral in Solanaceae, and that the SI mechanism was not regained within this family.
Therefore, for all breeding system models, we estimated a transition rate from $I$ to $C$  but not the reverse ($q_{CI}=0$).
% This is an important assumption because ancestral state reconstruction in the models might differ when polyploidy and breeding system are analyzed as independent in the Solanaceae tree.
%
%B: What happens if SC->SI is not fixed? Asking for a friend.  E: Surprisingly, Rosana reports, results are not turned on their head. 
%B: Whew, I was worried. Er, "friend" says thanks. R- Your welcome, by mistake I did that model with reversibility from C to I. I didn't run it for as long as the rest but preliminary ancestral reconstructions seem almost identical when you add that parameter than when you don't. 

\subsubsection{Ploidy, breeding system, and diversification}

Ploidy and breeding system might influence lineage diversification individually, but these two traits also have an intricate association (discussed in the Introduction).
Therefore, we considered several multi-state models that investigate the contribution of both traits and the allowable transitions between them.

The simplest classifies each species as either self-incompatible diploid $(ID)$, self-compatible diploid $(CD)$, or self-compatible polyploid $(CP)$; recall that self-incompatible polyploids do not occur (M16).
Each of these states may again be associated with different rates of speciation and extinction, and the allowable transitions are loss of SI within the diploid state (from $ID$ to $CD$), loss of SI via polyploidization (from $ID$ to $CP$), and polyploidization while SC (from $CD$ to $CP$).
% The total rate of loss of self-incompatibility, \ie transitions out of $ID$, is $q_{IC} + \rho_I$.
% E: Can we call it something other than q_IC?  It's a bit confusing to have the same parameter mean different things in different models.

As for the previous models of only one trait, we then allow diversification to be influenced by only a hidden trait (M17), or by both ploidy and breeding system as well as a hidden trait (M18--M20, with varying degrees of complexity in the hidden trait transitions).
We add this hidden trait layer on top of our three-state model \citep[similar to][]{caetano_2018, huang_2018}.
We also fit the analogous models but allowing for diploidization (M21--M25).

\subsubsection{Lumped models}

The models above allow us to assess the contributions of our two focal traits---ploidy and breeding system---to lineage diversification.
They do not, however, reveal whether it was ``worth it'' to include both traits in the analysis.
That is, they do not allow us to perform statistical model comparisons to test whether diversification heterogeneity is better explain by both traits than by either one of them.
This is because the models in the three sections above use different state spaces: $D/P$ versus $I/C$ versus $IC/CD/CP$.
Model comparisons work with the likelihood of the data given the model, but they are not valid when the data used differ from model to model.

In order to compare model fits for ploidy only versus breeding system only versus both traits, we use the technique of \citet{tarasov_2019} for lumping states together.
We use the state space of the $IC/CD/CP$ model but constrain the rate parameters to mimic the behavior of the single-trait models.
First we lump together $ID$ and $CD$ to form the diploid state, mimicking the $D/P$ model (M26).
Then we lump together $CD$ and $CP$ to form the self-compatible state, mimicking the $I/C$ model (M28).
We further add a hidden trait to each of these models (M27 and M29).

We do not include additional models with diploidization because this reverse transition renders the models non-lumpable \citep{tarasov_2019}.  % E: further explanation needed?

\subsubsection{Pathways to polyploidy}

Considering ploidy and breeding system together, there are two evolutionary pathways from SI diploid to SC polyploid \citep{brunet2001, robertson_2011}.
In the one-step pathway, the $CP$ state is produced directly from the $ID$ state when whole genome duplication disables SI.
In the two-step pathway, the $CD$ state is an intermediate: SI is first lost, and later the SC diploid undergoes polyploidization.
We quantify the relative contribution of these pathways to polyploidy in two ways, each using the maximum \textit{a posteriori} (MAP) estimates of rates from the simplest model that includes both traits (M16).

Both of our methods are based on a propagation matrix that describes flow from $ID$ to $CP$, as in \citet{robertson_2011}.
We insert an artificial division in the $CP$ state, so that one substate contains the $CP$ species that arrived via the one-step pathway and the other substate contains the $CP$ species that arrived via the two-step pathway.
We consider only unidirectional change along each step of the pathway in order to separate them into clear alternatives, and because in this family there is no support for regain of SI, and no strong support for diploidization (see below).

First, we consider only the rates of transitions between these states, placing them in the propagation matrix \myvec{Q}.
The matrix $\myvec{P} = \exp(\myvec{Q} t)$ then provides the probabilities of changing from one state to any other state after time $t$.
Closed-form solutions for the two pathway probabilities are provided in \citet{robertson_2011}.
Our results will differ from those of \citet{robertson_2011} because our transition rate estimates come from a dated phylogeny and a model that allows for state-dependent diversification.

Second, we consider not only transitions between states but also diversification within each state.
State-dependent diversification can change the relative contributions of the two pathways.
For example, if the net diversification rate is small for $CD$, the two-step pathway will contribute relatively less.
We therefore include the difference between speciation and extinction along the diagonal elements of the propagation matrix.
As before, matrix exponentiation provides the relative chance of changing from one state to any other state after time $t$.
In this case, however, these are not probabilities because diversification changes the number of lineages as time passes.
We can still use their ratios to consider the relative contribution of each pathway, though, analogous to the normalized age structure in a growing population. % R- here a reference would be nice  E: Does Jake have a favorite ref? Otherwise, I guess a basic ecology textbook.

% E & R todo ?: for the case with net div, write out the propagation matrix and equations for pathway contributions

% E: I don't think we need this section now that diploidization is in each of the sections above.  But I'm saving this text in case we want to put some of it in Results.
%--------------------------------------------------
% \subsubsection{Diploidization as an exploratory hypothesis}
%
% The models described above allow polyploidization but not its reverse.
% We explored the effects of allowing diploidization by adding a parameter $\delta$ for transitions from $P$ back to $D$ (in the $D/P$ and $D/P+A/B$ models) or from $CP$ back to $CD$ (in the $I/C$ and $I/C+A/B$ models); we did not allow transitions from $CP$ back to $ID$ because that would include regain of SI.
% Previous modeling approaches \citep{mayrose_2011} have argued against inferring diploidization rates when using ploidy data that comes from classifications based on chromosome number multiplicity or chromosome number change models like chromEvol \citep{mayrose_2010, glick2014, mayrose_2015, freyman_2017}, which assume only aneuploid change and polyploidization.
% Our ploidy classifications, however, when possible also incorporated information on chromosome multiplicity at the genus level and author conclusions within original studies with alternative sources of information (\eg geographic distribution, genus ploidy distribution).
% Since it is not known whether diploidization can be detected under these additional ploidy classifications, we fit models with and without diploidization in order to test whether the conclusions about diversification are sensitive to including diploidization.
% In our models, the diploidization parameter (or its absence, $\delta=0$), represents an opportunity to explore a possibly significant assumption that may not decisively affect interactions among ploidy, breeding system, and diversification.
%--------------------------------------------------

\subsection{Statistical inference}

\subsubsection{Model fitting}

Parameters for each of the ten models were estimated using RevBayes \citep{hoehna_2016}.
Code for analyses and key results are available as Supplementary Information. % FIXME more permanent than \url{https://github.com/roszenil/solploidy}.
We accounted for incomplete sampling in all analyses by setting the probability of sampling a species at the present to $651/3000$ \citep[using the method of][]{fitzjohn_2009} since the Solanaceae family has approximately 3,000 species \citep{solsource}.
For all models, we assumed that speciation and extinction parameters had log-normal prior distributions with means equal to the expected net diversification rate $(\text{number of taxa} / [2 \times \text{root age}])$ and standard deviation $0.5$.
Priors for parameters defining trait changes were assumed to be gamma distributed with parameters $k=0.5$ and $\theta=1$. 
For each model, a Markov chain Monte Carlo \citep[MCMC;][]{metropolis1953equation,Hastings1970} was run in the high-performance computational cluster at the Minnesota Supercomputing Institute, which allowed for 5,000 generations of burn-in and a minimum of 200,000 generations of MCMC for each of the  models. % FIXME see referee comment about reporting convergence, not runtime
Convergence and mixing of each MCMC chain was determined by ensuring the effective sample size of each parameter was over 200.

We report posterior distributions for the model parameters, and also for the compound parameters of net diversification ($\lambda - \mu$) and extinction fraction ($\mu / \lambda$) for each state.
Additionally, ancestral states at each node in the phylogeny were sampled jointly during the MCMC analyses.
Ancestral state reconstructions for all models show the maximum \emph{a posteriori} estimates of the marginal probability distributions for each of the 594 internal nodes for each of the 10 models (\cref{suppfigure:DPnodipasr,suppfigure:DPasr,suppfigure:DPnodipABasr,suppfigure:DPABasr,suppfigure:ICasr,suppfigure:ICABasr,suppfigure:IDCDCPnodipasr,suppfigure:IDCDCPasr,suppfigure:IDCDCPnodipABasr,suppfigure:IDCDCPABasr}). % FIXME: I'm confused about if these are joint or marginal?  A bit more explanation would help.

\subsubsection{Model selection}

We calculated the marginal likelihood for each of the models using fifty stepping stone steps under the methodology of \citet{xie_2010}, implemented in RevBayes \citep{hoehna_2016}.
Each stepping stone step was found by calculating at least 500 generations of burn-in followed by a total of 1,000 MCMC steps (\cref{table:marginallike}).

Using the marginal likelihood values, we calculated Bayes factors to argue five key biological and methodological questions:
\begin{enumerate}
\item Are hidden state necessary to add in this context?- Comparison of simple models against hidden state models
\item Is a second trait adding information to the diversification process?- Comparison of lumped models against IC/CD/CP models
\item Are diversification patterns only determined by hidden states and not the traits of interest?- Comparison of character independent models against hidden state 
\item Is there evidence for diploidization?- Comparison amongst models with and without diploidization
\item Do asymmetrical rates in the model affect the pattern of diversification? - Comparison amongst hidden states models with equal hidden rates and asymmetrical rates.
\end{enumerate}

Each model comparison is reported with a Bayes factor on the natural log scale: the comparison between models $M_0$ and $M_1$ is $BF(M_1,\ M_0) = \ln[ P(\mathbf{X} | M_1) - P(\mathbf{X} | M_0)]$.
There is `strong evidence'  for $M_1$ when this value is more than 2, `very strong' support when it is more than 6, and `extreme' support when it is more than 10 \citep{kass1995}.

\begin{figure}
\centering
\includegraphics[width=0.5\textwidth]{states} % FIXME: update numbers
\caption{
Character states used for each of the models.
Each species retained on the tree belonged to one of five possible categories, depending on whether ploidy and/or breeding system were known.
(A) The five categories and the number of species in each.
Abbreviations are $I$ for self-incompatible, $C$ for self-compatible, $D$ for diploid, $P$ for polyploid, $?$ for unknown.
Because polyploidization breaks this form of self-incompatibility, SI species with unobserved ploidy ($I?$) are taken to be diploid ($ID$), and polyploid species with unobserved breeding system ($?P$) are taken to be SC.
(B) Category groupings into states for each model type.
For example, there are 34 species that are SC with unknown ploidy. % FIXME: update number if necessary
In the $D/P$ models these are coded as either $D$ or $P$ (uncertain/consistent with either state), in the $I/C$ models they are coded as $C$, and in the $ID/CD/CP$ models they are coded as either $CD$ or $CP$.
In the hidden state models, all species were coded as consistent with both $A$ and $B$.
}
\label{figure:stateclassifications}
\end{figure}

\begin{suppfigure}
    \caption{ All models.  [This figure is provided as a separate, large-format page.] }
    \label{fig:allmodels}
\end{suppfigure}


%%% Below here is old tables.  I think the new versions should instead go in results.tex or a separate file. %%%

% \begin{table}
% \begin{tabular}{lcccccc} \toprule
% \begin{tabular}{@{}lcccccc@{}} \toprule
%\multicolumn{4}{r}{Models} \\ \cmidrule(r){3-5}
%Model                     & Ploidy & Diploidization & Breeding  & Hidden & Num        & Marginal       \\
%                          &        &                & System    & State  & Parameters & Log-Likelihood \\ \midrule
%1. D/P                    & Yes    & No             & No        & No     & 5          & -1193.66\\
%2. D/P +$\delta$          & Yes    & Yes            & No        & No     & 6          & -1182.93 \\
%3. D/P+A/B                & Yes    & No             & No        & Yes    & 10         & -1150.99\\
%4. D/P+A/B+$\delta$       & Yes    & Yes            & No        & Yes    & 11         & \textbf{-1145.69}\\
%5. I/C                    & No     & No             & Yes       & No     & 5          & -1194.80 \\
%6. I/C+A/B                & No     & No             & Yes       & Yes    & 10         & \textbf{-1155.37}\\
%7. ID/CD/CP               & Yes    & No             & Yes       & No     & 9          & -1345.87\\
%8. ID/CD/CP+$\delta$      & Yes    & Yes            & Yes       & No     & 10         & -1344.50\\
%9. ID/CD/CP+A/B           & Yes    & No             & Yes       & Yes    & 15         & -1303.55 \\ 
%10. ID/CD/CP+A/B+$\delta$ & Yes    & Yes            & Yes       & Yes    & 16         & \textbf{-1300.35} \\
%\bottomrule
%\end{tabular}
%\caption{
   % The ten models and their marginal likelihoods.
    %Values in bold are for the best models within each class that are comparable (see \cref{table:bayesfactors}).
    %Abbreviations are D: diploid, P: polyploid, I: self-incompatible, C: self-compatible, A: one state of hidden trait, B: other state of hidden trait, $\delta$: diploidization.}
%\label{table:marginallike}
%\end{table}

\begin{table}
\addtolength{\tabcolsep}{-3pt}
\begin{tabular}{|c|c|c|}
\bottomrule
\end{tabular}
% E: We will probably have to change the layout here because it's not technically a table.  But good enough for now.
\caption{
    Bayes factors for model comparisons.
    Each of the three boxes contains models that can be compared with one another, based on the character states they include (see \cref{figure:stateclassifications}B).
    Models are numbered in as \cref{table:marginallike}. % FIXME: they aren't, but it's a pain to change; change in prev table instead?
    Bayes factors are reported on the natural log scale, so numbers greater than $+2$ mean that the model in the row label has `positive' support relative to the model in the column label; numbers less than $-2$ mean that model in the column label is the preferred one.
    Conventional thresholds for `strong' and `very strong' support are 6 and 10, respectively.
    The best model in each set is written in bold.
    In each case, it is the most complex model of the set.
}
\label{table:bayesfactors}
\end{table}

% E: Could we include a figure of the traits on the tree?  At least as supp info. FIXME
