\section{Methods}

\subsection{Data}

Chromosome number data were obtained for all Solanaceae taxa in the the Chromosome Counts Database \citep[CCDB;][]{rice_2015}, and the ca.\ 14,000 records were cleaned semi-automatically using the CCDBcurator R package \citep{zenilferguson_2017}.
This large dataset includes the compilation of Solanaceae ploidy states from \citep{robertson_2011}.
Species were coded as either diploid (D) or polyploid (P).
For the majority of species, ploidy was assigned according to information from the original publications and the Kew Royal Botanic Gardens C-value DNA resource \citep{bennett_2005}.
For taxa without ploidy information but with information about chromosome number, we assigned ploidy based on the multiplicity of chromosomes within the genus.
For example, \textit{Solanum betaceum} did not include information about ploidy level but it has 24 chromosomes, so because $x=12$ is the base chromosome number of the \textit{Solanum} genus \citep{olmstead_2007}, we assigned \textit{S.~betaceum} as diploid. 
Species with more than one ploidy level were assigned the smallest most frequent ploidy level recorded. % E: Does this apply to the diploid-polyploid levels?  If so, we need to explain how it is also reconciled against the SI/SC value.  Or is this sentence just about multiple levels of polyploidy, which wouldn't need to be mentioned here?

Breeding system was scored as self-incompatible (SI) or self-compatible (SC) based on results hand-curated from the literature (as in \citealt{igic_2006, goldberg_2010, robertson_2011, goldberg_2012}).
Most species could unambiguously be coded as either SI or SC \citep{raduski_2012}.
Following previous work, we coded as SI any species with functional SI systems, even if SC or dioecy was also reported.
Dioecious species without functional SI were coded as SC.

To those existing data sets, we added some additional records for chromosome number and breeding system.
The Supplementary Information contains citations for the numerous original sources for all of the data. % todo: add number of refs when we have it
Synonymy followed Solanaceae Source \citep{solsource}.
Hybrids and cultivars were excluded because ploidy and breeding system are likely to be altered by domestication.
As in \citet{robertson_2011}, we examined closely the few species for which the merged ploidy and breeding system data indicated the presence of SI polyploids.
Although SI populations frequently contain some SC individuals and diploid populations frequently contain some polyploid individuals, in no case did we find individuals that were both SI and polyploid.
Because of this empirical observation and the functional explanation for whole genome duplication disabling gametophytic self-incompatibility \citep[reviewed in][]{ramsey_1998,stone_2002}, we consider only three observed character states: SI-D, SC-D, and SC-P.

% E: We should probably include some more details about how many polymorphic species and how the particularly tricky ones were dealt with.

Matching our character state data to the largest time-calibrated phylogeny of Solanaceae \citep{sarkinen_2013} yielded 595 species with ploidy and/or breeding system information on the tree.
Binary or three-state classification of ploidy and breeding system for the 595 taxa is summarized in \cref{table:stateclassifications}. % E: I think you're still updating the state coding?
We retained all of these species in each of the analyses below, because pruning away tips lacking breeding system in the ploidy-only analyses (and vice versa) would discard data that could inform the diversification models.
% E: Though someone might ask why tips with no state data were pruned.  Analyses took long enough on even this-size tree.

\subsection{Models}

% E: I think it would help to start with an "overall strategy" paragraph, maybe something like this.
To investigate the associations of these traits with lineage diversification, we fit six models from the BiSSE \citep{maddison_2007} family.
The simplest models consider ploidy only or breeding system only, as previous work has done. % add refs
We also consider the two traits simultaneously, to investigate whether the effect of one might override the effect of the other.
Additionally, to allow for the possibility that something other than these traits might affect diversification, we add models that include a hidden trait. % hisse ref
% E: And say something about including diploidization?  Will the motivation be in the Intro?
% E: And explain about the SC -> SI rate being fixed to 0, so we don't bury the explanation below? vs fixing ancestry?

We first considered only the ploidy trait, scoring each species in either the diploid state (D) or the polyploid state (P).
To allow for the influence of this binary trait on lineage diversification, we used the BiSSE model \citep{maddison_2007}.
% \cref{table:stateclassifications}
In a Bayesian framework, we obtained posterior probability distributions of speciation rates ($\lambda_D$, $\lambda_P$), extinction rates ($\mu_D$, $\mu_P$), net diversification rates ($r_D=\lambda_D-\mu_D,\ r_P=\lambda_P-\mu_P$), and relative extinction rates ($\nu_D = \mu_D / \lambda_D,\ \nu_D = \mu_D / \lambda_D$) for each state.
% as previously explored by \citet{mayrose_2011} % E: actually, they used bisse-ness/classe to allow for cladogenetic change
We also simultaneously estimated the polyploidization rate (the transition rate from D to P, $\rho$), and we allowed for diploidization (the transition rate from P to D, $\delta$).
We call this the ``ploidy only'' model. % E: not sure that we need to be this explicit, but try for now

Second, we considered ploidy and also a unobserved, hidden trait.
As discussed by \citet{beaulieu_2016}, the previous model might attribute diversification differences to the ploidy trait merely because there are no other explanatory factors in the model.
% As a second step, we fitted a hidden state speciation and extinction model (HiSSE, \citet{beaulieu_2016}) to evaluate whether the differences in diversification rates were found due to a hidden trait associated to polyploidy.
% As discussed by \citet{beaulieu_2016} BiSSE-like models suffer from a large type I error because they fail to account as part of the null hypothesis heterogeneous diversification rate changes  that do not depend on the trait of interest.
By including a hidden trait in a state speciation and extinction model, HiSSE-like models allow for heterogeneous background diversification while also parsing out the possible signal of diversification due to the trait of interest. 
Our second model thus has four states, with D and P subdivided by an unobserved binary trait with states A and B.
We therefore estimated the posterior probability distributions of four speciation rates ($\lambda_{D_A},\ \lambda_{D_B},\  \lambda_{P_A},\ \lambda_{P_B}$), four extinction rates ($\mu_{D_A},\ \mu_{D_B},\ \mu_{P_A},\ \mu_{P_B}$), and the corresponding four net diversification rates and relative extinction rates.
We again include transition due to polyploidization (rate $\rho$) and diploidization (rate $\delta$), and we assume that the changes between hidden states are symmetrical with rate $\alpha$.
We call this the ``ploidy and hidden trait'' model.

% E: I added the names of the rate parameters to these next models

In our third, ``breeding system only'' model, each species is scored as either self-incompatible (SI) or self-compatible (SC).
In this BiSSE analysis, state-dependent speciation rates ($\lambda_I$, $\lambda_C$) and extinction rates ($\mu_I$, $\mu_C$) are estimated, with the corresponding net diversification and relative extinction rates.
The rate for transitions from SI to SC is $q_{IC}$, and regain of SI is prohibited, as explained above.
This is the same as the analysis of \citet{goldberg_2010}.

The fourth model considers both breeding system and a hidden trait, again using a HiSSE strategy to test whether diversification shifts unrelated to the focal trait might be attributed to it by the previous BiSSE analysis.
In this ``breeding system and hidden trait'' model, there are four speciation rates ($\lambda_{I_A},\ \lambda_{I_B},\  \lambda_{C_A},\ \lambda_{C_B}$), four extinction rates ($\mu_{I_A},\ \mu_{I_B},\ \mu_{C_A},\ \mu_{C_B}$), and correspondingly four rates of net diversification and relative extinction.
Transition rates are again $q_{IC}$ for loss of SI and $\alpha$ for changes in the hidden state.

% E: I adjusted the rate names a little.  Does this seem okay?
Our fifth model investigates the link among diversification, breeding system, and ploidy simultaneously.
There are three possible states: self-incompatible diploids (ID), self-compatible diploids (CD), and self-compatible polyploids (CP); self-incompatible polyploids are not present in this family, as explained earlier.
This ``ploidy and breeding system'' MuSSE model \citep{fitzjohn_2012} has six rates for speciation and extinction ($\lambda_{ID},\ \mu_{ID},\ \lambda_{CD},\ \mu_{CD},\ \lambda_{CP},\ \mu_{CP}$).
It also has four rates for transitions:
polyploidization of SI diploids (transition from ID to CP at rate $\rho_I$),
polyploidization of SC diploids (transition from CD to CP at rate $\rho_C$),
diploidization, which does not restore SI (transition from CP to CD at rate $\delta_C$),
and loss of SI without a change in ploidy (transition from ID to CD at rate $q_{IC}$).

Finally, the sixth model considers a hidden trait along with ploidy and breeding system.
This again provides a means to account simultaneously for diversification rates linked to ploidy and breeding system while also allowing for the presence of heterogeneity from unknown sources.
(A similar approach was used for a different variant of BiSSE by \citealt{huang_2018,caetano_2018}.)
The full model would have 26 parameters, but our goal is to look for diversification rate differences so we fit a simplified version with 17 parameters by fixing the transition rates among hidden states to be equal with parameter $\alpha$, and the transition rates between breeding system and ploidy states as defined in the previous model ($\rho_I$, $\rho_C$, $\delta_C$,  $q_{IC}$) independent of the hidden state.
We estimated twelve speciation and extinction rates ($\lambda_{ID_A},\ \mu_{ID_A},\ \lambda_{CD_A},\ \mu_{CD_A},\ \lambda_{CP_A},\ \mu_{CP_A},\ \lambda_{ID_B},\ \mu_{ID_B},\ \lambda_{CD_B},\ \mu_{CD_B}, \lambda_{CP_B},\ \mu_{CP_B}$), and the net diversification and relative extinction rates associated with them.

All the models were performed using RevBayes \citep{hoehna_2016} software that performs Bayesian inference via MCMC using Metropolis-Hastings algorithm.
A correction for sampling bias was done in all models by assuming that Solanaceae family has approximately 3,000 species ($s=595/3000$) as the Solanaceae Source project indicates \citep{solsource}.
In all models, speciation and extinction parameters used log-normal prior distributions that with mean the expected net diversification rate $ ( \frac{(number of taxa)/2}{root age})$ and standard deviation $0.5$.
Prior distributions for parameters defining cladogenetic changes were gamma distributed with parameters $k=0.5$ and $\theta=1$. 
MCMC was performed for 96 hours in the cluster at Minnesota Supercomputing Instute which allowed for 5,000 generation of burn-in and a minimum of 200,000 generations of MCMC for each of the 6 models.
Convergence of the MCMC was tested using R package coda (ref) and software Tracer (ref) to assess convergence and mixing (see supplementary information).


\begin{table}
\begin{tabular}{@{}llccc@{}} \toprule
\multicolumn{4}{r}{Models} \\ \cmidrule(r){3-5}
Type & Total of & BiSSE/HiSSE & BiSSE/HiSSE & MuSSE/MuHiSSE\\ 
 &Taxa &  Polyploidy & Breeding System &  \\ \midrule
Diploid Self Compatible & 152 & 0 &  0 & 0 \\
Diploid Self Incompatible& 97 & 0  & 1 & 2\\
Diploid with unknown breeding system & 219 & 0 & (0,1) & (0,2) \\
Polyploid & 81 & 1& 0 & 1 \\
Unknow ploidy and self compatible& 34 & (0,1)& 0 & (0,1) \\ 
Unknown ploidy and self incompatible & 12 & 0 & 1 & 2 \\ \bottomrule
\end{tabular}
\caption{Binary and three state classifications for 595 taxa with ploidy and/or breeding system data. The number of taxa in the t he sample was maximize by including tips with only ploidy or only breeding system and assigned them as uncertain in the unknown character.}
% E: but P are not SI
\label{table:stateclassifications}
\end{table}
