\section{Methods}

\subsection{Data}

%B: IMO: The data subsection would be better organized with parts, (1) ploidy, (2) breeding system, (3) tree, (4) models, (5) model selection and statistical inference.
% E: I changed the subsectioning, but to a different arrangement: (1) Data, (2) Models, (3) Statistical inference.  With subsubsections as necessary.

Chromosome number data were obtained for all Solanaceae taxa in the Chromosome Counts Database \citep[CCDB;][]{rice_2015}, and the ca.~14,000 records were curated semi-automatically using the \mbox{CCDBcurator} R package \citep{rivero_2019}.
CCDB contains records from original sources that have multiple complex symbol patterns denoting multivalence, or irregularites of chromosome counts.
After a first round of automatic cleaning, we examined results by hand and corrected records as necessary.
Our hand-curated records were also contrasted against the ploidy dataset from \citet{robertson_2011}, original references therein, and against ploidy data in the C-value DNA dataset from \citet{bennett_2005}.
By comparing three different sources of information, we were able to code taxa as diploid, $D$, or polyploid, $P$.
For the majority of species, ploidy was assigned according to information from the original publications included in the  C-value DNA dataset \citep{bennett_2005}.
For taxa without ploidy information but with information about chromosome number, we assigned ploidy based on the multiplicity of chromosomes within the genus/family, or based on SI/SC classification.
For example, \textit{Solanum betaceum} did not have information about ploidy level, but it has 2n=24 chromosomes, and  $x=12$ is the base chromosome number of the genus \textit{Solanum} \citep{olmstead_2007}, so we assigned \textit{S.~betaceum} as diploid. 
Additionally, because of the absence of SI polyploids (explained above and below), species known to be SI could be scored as diploid.
Species with more than one ploidy level were assigned the most frequent ploidy level recorded or the smallest ploidy in case of frequency ties.

Breeding system states were scored as self-incompatible, $I$, or self-compatible, $C$, based on results curated from the literature  \citep[as compiled in][]{igic_2006, goldberg_2010, robertson_2011, goldberg_2012} and original experimental crosses (B.I. and E.E.G., unpub. data). %IM suggestion %B:  I returned the aside about original data. I don't think Itay knows that we performed original crosses (many), and they are incorporated in the data, without being present in "the literature" 
%was originally: Breeding system states were scored as self-incompatible, $I$, or self-compatible, $C$, based on results curated from the literature and original experimental crosses \citep[as compiled in][]{igic_2006, goldberg_2010, robertson_2011, goldberg_2012}.
Most species could unambiguously be coded as either $I$ or $C$ \citep{raduski_2012}.
Following previous work, we coded any species with a functional SI system as $I$, even if SC or dioecy was also reported.
Dioecious species without functional SI were coded as $C$.
% E: I am trying to stick with I/C when talking about the state values, but using SI/SC when talking about the meaning of the trait (because this what people use normally). 
%R- Okay- I was getting confused as well thanks for clarifying

Resolution of taxonomic synonymy followed Solanaceae Source \citep{solsource}. 
Hybrids and cultivars were excluded because ploidy and breeding system can be affected by artificial selection during domestication.
Following the reasoning outlined in \citet{robertson_2011}, we closely examined the few species for which the merged ploidy and breeding system data indicated the presence of self-incompatible polyploids.
Although SI populations frequently contain some SC individuals, and diploid populations frequently contain some polyploid individuals, in no case did we find convincing data for a naturally occurring SI polyploid population  \citep[discussed in][]{robertson_2011}.
%B rm: The single instance of an SI polyploid individual appears to be an allopentaploid hybrid of \textit{Solanum oplocense} Hawkes x \textit{Solanum gourlayii} Hawkes, reported by \citet{camadro_1981}.
% Under exceedingly rare circumstances, it is possible for polyploids containing multiple copies of S-loci to remain SI, so long as they express a single allele at the S-locus \citep[discussed in][]{robertson_2011}.
Because of the resulting absence of polyploid SI populations, as well as the functional explanation for polyploidy disabling gametophytic SI systems with non-self recognition (see the Introduction), we consider only three observed character states: self-incompatible diploids $(ID)$, self-compatible diploids $(CD)$, and self-compatible polyploids $(CP)$.

Matching our character state data to the largest time-calibrated phylogeny of Solanaceae \citep{sarkinen_2013} yielded 651 species with ploidy and/or breeding system information on the tree.
% Fixed!
Of these, 368 had information for both states.
The number of species in each combination of states is summarized in \cref{figure:stateclassifications}A and in \cref{suppfigure:stateclassifications}.
We retained all 651 species in each of the analyses below because pruning away tips lacking breeding system in the ploidy-only analyses (and vice versa) would discard data that could inform the diversification models.
A total of 372 taxa without any information about breeding system or ploidy were excluded.
%Tips without trait data might not improve point estimation of diversification parameters while increasing uncertainty in the estimates of trait linked diversification
% R-This is very important but I have removed it because I think it is a full on paper by its own
%Including this many more species would have prohibitively slowed our analyses, especially those implementing the most complex models.

The Supplementary Information contains citations for the numerous original data sources. % FIXME (and say here how many refs?) %B: no need for an additional counting and updating headache?
% R- There is few data that I added so I can find those references and send them to you, so it would be simply the same references that you previously had plus another 20-25 I think
The Dryad archive contains the data and tree files used for analyses. % FIXME
% \url{https://github.com/roszenil/solploidy}  E: we should use dryad instead, because github repos are not guaranteed to last
%This fixme happens after publications

\subsection{Models}

In order to test our hypotheses about lineage diversification and trait macroevolution, we fit twenty-nine state-dependent speciation and extinction models \citep[BiSSE, MuSSE, HiSSE;][]{maddison_2007, fitzjohn_2012, beaulieu_2016}.
SSE models contain parameters that describe per-lineage rates of speciation and extinction, specific to each character state (denoted $\lambda$ and $\mu$, respectively, with subscripts to indicate the state), along with rates of transitions between states (denoted $\rho$ for polyploidization, $\delta$ for diploidization, and $q_{IC}$ for loss of self-incompatibility).
The full set of models and all their rate parameters are detailed in \cref{fig:allmodels}.
Here, we summarize how each model allows us to assess whether diversification is best explained by variation in ploidy, breeding system, their combination, or some unknown factor.

\subsubsection{Ploidy and diversification}

We first employed a model (labeled M1), previously used by \citet{mayrose_2011}, with each species classified as diploid $(D)$ or polyploid $(P)$.
Although this model can be powerful in studies of trait evolution, it is prone to incorrectly reporting that a trait is associated with diversification differences \citep{maddison_2015, rabosky_2015}.
We therefore define several models that incorporate additional forms of diversification rate heterogeneity.
The second ploidy model (M2) includes a binary hidden trait that subdivides each observed state.
In this trait-independent model known as CID \citep{beaulieu_2016}, hidden traits can affect diversification but the observed traits do not.
Comparing M1 and M2 allows us to test whether diversification rate heterogeneity is better explained by ploidy or by some unknown factor.

We fit three models in which both ploidy and a hidden trait could influence diversification (M3--M5).
These models differ in whether transitions between the hidden states are symmetric (M3) or asymmetric(M4), and whether the polyploidization rate depends on the hidden state(M5). %WF suggestion
Comparing M1--M5 (as shown in \cref{table:bayesfactors}) allows us to test whether ploidy is associated with diversification differences on top of the differences potentially explained by an unknown factor.
We further fit the analogues of these five models but including a rate parameter $\delta$ for transitions from polyploid to diploid (M6--M10).
These comparisons allow us to assess whether our conclusions about ploidy and diversification are robust to the possibility of diploidization.

\subsubsection{Breeding system and diversification}

We propose five breeding system models following the same logic as the ploidy models above.
Under the simplest breeding system and diversification model (M11), species are classified as self-incompatible $(I)$ or self-compatible $(C)$.
This is the same model as in the analysis presented in \citet{goldberg_2010} but with an updated phylogeny \citep{sarkinen_2013} and a larger aggregated dataset.
We then add models to allow diversification to be influenced by only a hidden trait (M12), or by both breeding system and a hidden trait (M13--M15, with varying degrees of complexity in the hidden trait transitions \cref{table:M11M15}).
Similar models were used by \citet{freyman_2019} to study diversification in Onagraceae.

Self-incompatibility is homologous in all Solanaceae species in which S-alleles have been cloned and controlled crosses performed.
All species sampled to date possess a non-self recognition, RNase-based gametophytic self-incompatibility \citep[shared even with other euasterid families;][]{ramanauskas_2017}.
Furthermore, species that are distantly related within this family carry closely-related alleles, with deep trans-specific polymorphism at the locus that controls the SI response \citep{ioerger_1990, igic_2006}.
Thus, there is strong evidence in Solanaceae that the $I$ state is ancestral in the family, and that the SI mechanism was not regained.
For all breeding system models, we estimated a transition rate from $I$ to $C$  but not the reverse ($q_{CI}=0$).
% This is an important assumption because ancestral state reconstruction in the models might differ when polyploidy and breeding system are analyzed as independent in the Solanaceae tree.
%
%B: What happens if SC->SI is not fixed? Asking for a friend.  E: Surprisingly, Rosana reports, results are not turned on their head. 
%B: Whew, I was worried. Er, "friend" says thanks. R- Your welcome, by mistake I did that model with reversibility from C to I. I didn't run it for as long as the rest but preliminary ancestral reconstructions seem almost identical when you add that parameter than when you don't. 

\subsubsection{Ploidy, breeding system, and diversification}

Ploidy and breeding system might influence lineage diversification individually, but these two traits also have an intricate association (discussed in the Introduction).
Therefore, we considered several multi-state models that investigate the contribution of both traits and the allowable transitions between them.

The simplest model (M16) classifies each species as either SI diploid $(ID)$, SC diploid $(CD)$, or SC polyploid $(CP)$; recall that SI polyploids do not occur.
Each of these states may again be associated with different rates of speciation and extinction, and the allowable transitions are loss of SI within the diploid state (from $ID$ to $CD$), loss of SI via polyploidization (from $ID$ to $CP$), and polyploidization while SC (from $CD$ to $CP$).
% The total rate of loss of self-incompatibility, \ie transitions out of $ID$, is $q_{IC} + \rho_I$.
% E: Can we call it something other than q_IC?  It's a bit confusing to have the same parameter mean different things in different models.
% R: In the lumped models I call them something different q_0 and rho_0 because they are different parameters but I don't think we should explain.
As for the previous models of only one trait, we then allow diversification to be influenced by only a hidden trait (M17), or by ploidy, breeding system, and a hidden trait (M18--M20) with varying degrees of complexity in the hidden trait transitions
\citep[similar to][]{caetano_2018, huang_2018}. %IM suggestion
We also fit the analogous models but allowing for diploidization (M21--M25).

\subsubsection{Lumped models}

The models described so far allow us to assess the contributions of our two focal characters---ploidy and breeding system---to lineage diversification, but they do not reveal whether it is valuable to include both characters in the analysis.
To answer this question with statistical model comparisons requires comparing the likelihood of the data given each model.
This is impossible for the ploidy and breeding system models presented so far, however, because the data are different for the different models: they use either the $D/P$ or the $I/C$ or the $ID/CD/CP$ state spaces (see figure \cref{figure:stateclassifications} for state classifications).
Therefore, the use of different data results in incomparable models.

In order to compare fits of ploidy-only \vs breeding system-only \vs combined trait models, we use the technique of `lumping' states together \citep{tarasov_2019}.
We use the state space of the $ID/CD/CP$ model but constrain the rate parameters to mimic the behavior of the single-trait models.
Lumping states requires that the transition rates from the lumped state to the singular state be equal \citep{tarasov_2019}.
First, we lump together $ID$ and $CD$ to form the diploid state, mimicking the $D/P$ model (M26). 
Proposing a lumped ploidy model by aggregating $ID$ and $CD$ requires forcing the rate of polyploidization from $ID$ and $CD$ to $CP$ to be equal (i.e. $\rho_0=\rho_I=\rho_c$), but also requires
assuming that the rates of speciation and extinciton for $ID$ and $CD$ to be equal.
Therefore, we define the new parameters $\lambda_D$ and $\mu_D$  that are the same for each of the two diploid states $ID$ and $CD$. %IM suggestion, describe more the lumping
We used the same procedure to lump together $CD$ and $CP$ to form the self-compatible state, mimicking the $I/C$ model (M28).
In this particular case, the rate from $CD$ to $CP$ back to $ID$ is zero and equal for both, so the model is lumpable. 
However, to fully mimic the breeding system model, we assume that the rate of selfing is equal (i.e. $q_0=q_{IC}=\rho_I$) and the rates of speciation and extinction for both $CD$ and $CP$ are the same (new parameters $\lambda_C$ and $\mu_C$).
We further add a hidden character to each of these models (M27 and M29), and then compare this group of models (\cref{table:lumped}).

We do not include additional models with diploidization because this reverse ploidy transition renders the models non-lumpable.
When including diploidization, transitions from $CP$ to $CD$ are at rate $\delta$ but transitions from $CP$ to $ID$ do not occur.
Because $ID$ and $CD$ would be lumped to mimic the $D/P$ model, this model is non-lumpable when $\delta \ne 0$.
Thus, we can compare models to test whether it is advantageous to include both traits, but only when ignoring diploidization.
% E: Below was all good, but it seemed like more detail than necessary.
% Then we define a single parameter $q_0$ going into the lumped state $(CD,CP)$ as the new rate from SI to SC.
% This defines the lumped model M28 (\cref{figure:lumped}E) that is equivalent to M1 but comparable to M16.
% We repeat this process to create the lumped model M26 for $D/P$ by restricting the polyploidy rate to be $rho_0$ out of the lumped state $(ID,CD)$.
% When including diploidization, however, transitions from $CP$ to $CD$ are at rate $\delta$ but transitions from $CP$ to $ID$ do not occur.
% Since $\delta \neq 0$, the models with diplodization cannot be lumped to be compared to the three-state models.

\subsubsection{Pathways to polyploidy}

Considering ploidy and breeding system together, there are two evolutionary pathways from SI diploid to SC polyploid \citep{brunet2001, robertson_2011}.
In the one-step pathway, the $CP$ state is produced directly from the $ID$ state when whole genome duplication disables SI.
In the two-step pathway, the $CD$ state is an intermediate: SI is first lost, and later the SC diploid undergoes polyploidization.
We quantify the relative contribution of these pathways to polyploidy in two ways, each using the median estimates of rates from the simplest model that includes both traits (M16). 
Our results differ from those of \citet{robertson_2011} in part because our approach relies on a dated phylogeny and models that allow for state-dependent diversification. %IM suggestion

Both of our methods are based on a propagation matrix that describes flow from $ID$ to $CP$, as in \citet{robertson_2011}.
We insert an artificial division in the $CP$ state, so that one sub-state contains the $CP$ species that arrived via the one-step pathway and the other substate contains the $CP$ species that arrived via the two-step pathway.
We consider unidirectional change along each step of the pathway in order to separate them into clear alternatives, and because in this family there is no support for regain of SI, and no strong support for diploidization (see below).
First, we consider only the rates of transitions between these states, placing them in the propagation matrix \myvec{Q}.
The matrix $\myvec{P} = \exp(\myvec{Q} t)$ then provides the probabilities of changing from one state to any other state after time $t$.
Closed-form solutions for the two pathway probabilities are provided in \citet{robertson_2011}.
Second, we consider not only transitions between states but also diversification within each state.
State-dependent diversification can change the relative contributions of the two pathways.
For example, if the net diversification rate is small for $CD$, the two-step pathway will contribute relatively less.
We therefore include the difference between speciation and extinction along the diagonal elements of the propagation matrix.
As before, matrix exponentiation provides the relative chance of changing from one state to any other state after time $t$.
The calculations of the propagation matrix are not probabilities because diversification changes the number of lineages as time passes. 
We can still use ratios, however, to consider the relative contribution of each pathway, analogous to the normalized age structure in a growing population \citep{leslie_1945}.
% E & R todo ?: for the case with net div, write out the propagation matrix and equations for pathway contributions
% R- couldn't solve to get the exact solutions turns out it is really hard.

\subsection{Statistical inference}

\subsubsection{Model fitting}

Parameters for each of the 29 models were coded as graphical models and Bayesian statistical inference was performed with RevBayes \citep{hoehna_2016}.
Scripts for analyses and key results are available in Dryad. %TODO
We accounted for incomplete sampling in all analyses by setting the probability of sampling a species at the present to $651/3000$ \citep[using the method of][]{fitzjohn_2009} since the Solanaceae family has approximately 3,000 species \citep{solsource}.
For all models, we assumed that speciation and extinction parameters had log-normal prior distributions with means equal to the expected net diversification rate $(\text{number of taxa} / [2 \times \text{root age}])$ and standard deviation $0.5$.
Priors for parameters defining trait changes were assumed to be gamma distributed with parameters $k=0.5$ and $\theta=1$. 
For each model, a Markov chain Monte Carlo \citep[MCMC;][]{metropolis1953equation,Hastings1970} was run in the high-performance computational cluster at the Minnesota Supercomputing Institute, which allowed for 5,000 generations of burn-in and a minimum of 200,000 generations of MCMC for each of the models.
Convergence and mixing of each MCMC chain was determined by ensuring the effective sample size of each parameter was over 200.

We report posterior distributions for the net diversification parameters ($\lambda - \mu$)  in \cref{figure:netdivall} and full results of all parameters of the diversification models in \cref{suppfigure:DPnodip,suppfigure:DPnodipAB,suppfigure:IC,suppfigure:ICAB,suppfigure:IDCDCPnodip,suppfigure:IDCDCPnodipAB}.
Additionally, ancestral states at each node in the phylogeny were sampled jointly during the MCMC analyses every 100 generations. 
%R: Itay says remove but some reviewers were interested in the distribution of the states. I say leave we don't need to discuss. I'll mention briefly. %B: I am not sure what this comment refers to (up or down?), but I'm not touching.
Ancestral state estimations for all models show the maximum \emph{a posteriori} estimates of the marginal probability distributions for each of the 650 internal nodes for each of the models in \cref{figure:netdivall}. (\cref{suppfigure:DPnodipasr,suppfigure:DPnodipABasr,suppfigure:ICasr,suppfigure:ICABasr,suppfigure:IDCDCPnodipasr,suppfigure:IDCDCPnodipABasr}).

\subsubsection{Model selection key questions}

We calculated the marginal log-likelihood for each of the models using fifty stepping stone steps under the methodology of \citet{xie_2010}, implemented in RevBayes \citep{hoehna_2016}.
Each stepping stone step was found by calculating at least 500 generations of burn-in followed by a total of 1,000 MCMC steps (\cref{table:bayesfactors}).

Using the marginal likelihood values, we calculated Bayes factors to answer five key biological and methodological questions:
\begin{enumerate}
\item Are diversification patterns only determined by hidden states and not the traits of interest?---Comparison of character independent models against hidden state (\cref{table:bayesfactors,table:M11M15,supptable:M6M10,supptable:M16M20,supptable:M21M25}).
\item Are hidden states necessary to explain diversification rate heterogeneity?---Comparison of simple models against hidden state models (\cref{supptable:testaddhidden}).
\item Does a second focal trait add information about the diversification process?---Comparison of lumped models against IC/CD/CP models (\cref{figure:lumped,table:lumped,suppfigure:lumpedDP,suppfigure:lumpedIC}).
\item Are conclusions robust to assumptions about hidden state transitions?---Comparison amongst hidden states models with equal hidden rates and asymmetrical rates (\cref{supptable:asymmetry,suppfigure:asymmetric}).
\item Is there evidence for diploidization?---Comparison amongst log-scalemodels with and without diploidization (\cref{supptable:testdiploidization,suppfigure:alldip}).
\end{enumerate}

Each model comparison is reported with a Bayes factor on the natural log-scale: the comparison between models $M_0$ and $M_1$ is $K=ln(BF(M_1,\ M_0)) = \ln[ P(\mathbf{X} | M_1) - P(\mathbf{X} | M_0)]$.
There is `strong evidence' for $M_1$ when this value is more than 10, moderate support if the value is more than 1, and no evidence if the value is between -1 and 1.
If the value of $K$ is negative the evidence goes towards $M_0$ \citep{kass1995}.


\begin{table}
\addtolength{\tabcolsep}{-3pt}
\begin{tabular}{|l|r|c|c|c|c|l|}
\hline
Model                & \begin{tabular}[c]{@{}l@{}}Marginal\\ log-likelihood\end{tabular} & M2    & M3     & M4    & M5     & Evidence                                                                                     \\ \hline
M1. D/P              & -1283.76                                                          & 59.90 & 49.23  & 60.48 & 58.82  & \begin{tabular}[c]{@{}l@{}}Every model strongly \\ preferred over M1\end{tabular}            \\ \hline
\textbf{M2. CID D/P} & \textbf{-1223.86}                                                 &       & -10.66 & 0.579 & -1.079 & \begin{tabular}[c]{@{}l@{}}Strong preference for \\ M2 over M1 and M3\end{tabular}              \\ \hline
M3. D/P+A/B          & -1234.52                                                          &       &        & 11.24 & 9.58   & \begin{tabular}[c]{@{}l@{}}Asymmetric rates strongly\\ preferred over symmetric\end{tabular}   \\ \hline
\textbf{M4. D/P+A/B asym}     & \textbf{-1223.28}                                                          &       &        &       & -1.658 & \begin{tabular}[c]{@{}l@{}}Moderate evidence for\\ only asymmetric hidden rates\end{tabular} \\\hline
M5. D/P+A/B all asym & -1224.93                                                          &       &        &       &        &          \\ \hline                                                                                   
\end{tabular}
\caption{Bayes factors for ploidy only models without diplodization in log-scale.  Numbers smaller than -1 indicate moderate preference for the model listed in the row, and values larger than 1 indicate moderate preference for the model listed in the respective column .  Conventional threshold for `strong' preference is an absolute value larger than 10.  The character independent model (M2, bold) is strongly preferred over the ploidy only model (M1) or a hidden state model with symmetric rates (M3). Moderate support exists for the model with ploidy and asymmetric hidden states (M4, bold) over the rest of the ploidy only models.}
\label{table:bayesfactors}
\end{table}

% B: M111->M11
\begin{table}
\addtolength{\tabcolsep}{-3pt}
\begin{tabular}{|l|r|c|c|c|c|l|}
\hline
Model                      & \begin{tabular}[c]{@{}l@{}}Marginal\\ log-likelihood\end{tabular} & M12   & M13   & M14   & M15   & Evidence                                                                                       \\ \hline
M11. I/C                   & -1309.07                                                          & 41.13 & 38.59 & 61.34 & 61.40 & \begin{tabular}[c]{@{}l@{}}Every model strongly \\ preferred over M11\end{tabular}            \\ \hline
M12. CID I/C               & -1267.93                                                          &       & -2.53 & 20.21 & 20.37 & \begin{tabular}[c]{@{}l@{}}Models with asymmetric rates\\ are preferred over  M12\end{tabular} \\ \hline
M13. I/C+A/B               & -1270.47                                                          &       &       & 22.75 & 22.80 & \begin{tabular}[c]{@{}l@{}}Asymmetric rates strongly\\ preferred over symmetric\end{tabular}   \\ \hline
\textbf{M14. I/C+A/B asym} & \textbf{-1247.72}                                                 &       &       &       & 0.05 & No evidence                                                                                    \\ \hline
\textbf{M15. I/C+A/B all asym }     & \textbf{ -1247.66}                                                          &       &       &       &       &       \\ \hline                                                                                        
\end{tabular}
\caption{Bayes factors for breeding system only models in log-scale.  Numbers smaller than -1 indicate moderate preference for the model listed on the row, and values larger than 1 indicate moderate preference for the model in the respective column.  Conventional threshold for `strong' preference is an absolute value larger than 10. Moderate to strong preference exist for models M14 and M15 which include a hidden states with asymmetrical rates.}
\label{table:M11M15}
\end{table}

% E: Could we include a figure of the traits on the tree?  At least as supp info. %FIXME

\begin{table}[]
\addtolength{\tabcolsep}{-3pt}
\begin{tabular}{|l|c|c|c|c|}
\hline
Model                       & \begin{tabular}[c]{@{}l@{}}Marginal\\ log-likelihood\end{tabular} & Comparison  & K=ln(BF($M_{i}$,$M_{ii}$)) & \begin{tabular}[c]{@{}l@{}}Preferred\\ Model (Evidence)\end{tabular} \\ \hline
M26. Lumped D/P             & -1463.22                                                          & M26 vs. M16 & 4.10            & M16 (Moderate)                                                       \\
\textbf{M16. ID/CD/CP}      & \textbf{-1459.11}                                                 &             &                 &                                                                      \\ \hline
M27. Lumped D/P+A/B         & -1417.67                                                          & M27 vs. M23 & 3.678           & M23 (Moderate)                                                       \\
\textbf{M23. ID/CD/CP +A/B} & \textbf{-1414.00}                                                 &             &                 &                                                                      \\ \hline
M28. Lumped I/C             & -1458.41                                                          & M28 vs. M16 & -0.69           & No evidence                                                          \\
M16. ID/CD/CP      & \textbf{-1459.11}                                                 &             &                 &                                                                      \\ \hline
M29. Lumped I/C+ A/B        & -1416.60                                                          & M29 vs. M23 & 2.60            & M23 (Moderate)                                                       \\
\textbf{M23. IC/CD/CP+A/B}  & \textbf{-1414.00}                                                 &             &                 &                                                                      \\ \hline
\end{tabular}
\caption{Testing the addition of a focal trait to a binary state model via Bayes factors (log-scale). Numbers smaller than -1 indicate moderate preference for the model in the row, and values larger than 1 indicate moderate preference for the model in the respective column.  Conventional threshold for `strong' preference is an absolute value larger than 10. Three-state models (M16, M23, bold) are moderately preferred over two-state models . Moderate evidence exists towards the inclusion of breeding systems in ploidy models (M16, M23, bold). Moderate to no evidence towards the  inclusion of ploidy in breeding system models is indicated by the comparisons of models M28 \vs M16 and M29 \vs M23 where the difference is the inclusion of ploidy information.}
\label{table:lumped}
\end{table}

\begin{figure}
\centering
%\includegraphics[width=0.5\textwidth]{fig1.pdf}
\caption{
Character states used in the models.
(A) Each species retained on the phylogeny belonged to one of five possible categories, depending on whether ploidy and/or breeding system were known. 
Number of species in each category is indicated; for example, 70 species are self-compatible with unknown ploidy.
Character state abbreviations are: $I$ for self-incompatible, $C$ for self-compatible, $D$ for diploid, $P$ for polyploid, $?$ for unknown.
Because polyploidization breaks this form of self-incompatibility, self-incompatible species with unobserved ploidy ($I?$) are taken to be diploid ($ID$), and polyploid species with unobserved breeding system ($?P$) are taken to be SC ($CP$).
(B) Category groupings into states for each model class.
In the ploidy-only models (M1- M10), states are coded as $D$ \& $P$ when uncertain/consistent with either state; in the breeding system-only models (M11-M15) such states are coded as $C$; in the ploidy and breeding system models (M16-M29), they are coded as $CD$ \& $CP$.
In the hidden-trait models, all species could take on either of two `hidden' character states. %IM: #33 
Two species, \emph{Lycium californicum} and \emph{Solanum bulbocastanum}, are simultaneously $ID$ and $CP$, and by adding them the sample adds to the total of 651 taxa used for analyses.
}\label{figure:stateclassifications}
\end{figure}

\begin{figure}
%\includegraphics[width=\textwidth]{fig2.pdf} 
\caption{Net diversification rates for SSE models of focal traits with or without hidden state. 
(A) Ploidy only model (M1) showing higher net diversification linked to diploid state $D$ compared to polyploid state $P$.
(B) Ploidy with hidden states model (M4) showing that the net diversification is higher for hidden state $B$ (lighter colors) compared to hidden state $A$ (darker colors) and both diploid and polyploid states within each hidden traits have overlapping net diversification rates.
(C) Breeding system only model (M9) showing higher net diversification linked to self-incompatible $I$ state compared to self-compatible state $C$.
(D) Breeding system with hidden states model (M11) showing diversification differences in both hidden states (light \vs. dark colors) and little to no overlapping in between self-compatible \vs self-compatible states.
(E) Ploidy and breeding system model (M13) showing higher net diversification linked to self-incompatible diploid state $ID$ compared to both self-compatible states despite ploidy level ($CD, CP$).
(F) Ploidy, breeding system, and hidden states model (M16) showing a similar pattern that panel (E) within each hidden state $A$ and $B$. For hidden state $B$ there is a larger overlap of net diversification between states $CD_B$ and $ID_B$.}
\label{figure:netdivall}
\end{figure}

\begin{figure}
%\includegraphics[width=\textwidth]{fig3.pdf} %lumped.pdf
\caption{Comparing two and three-state models using lumped models. (A) Ploidy model only (M1) where data enter as binary $D$ and $P$. 
(B) Lumped model for ploidy (M26) where data are the three-state values ($ID,CP,CD$) but results are equivalent to model M1.  
(C) Ploidy and breeding system model (M16) where  data enter as the three-state values. Models M26 and M16 are comparable whereas M1 and M16 are not.
(D) Breeding system only model (M11) where data are entered as binary $I$ and $C$. 
(E) Lumped model for breeding system (M28) where data are the three-state values ($ID,CP,CD$) but results are equivalent to model M11. Model M26 can be compared with model M16 from panel (C).
Model comparisons are done via Bayes factors and results are shown in \cref{table:lumped}}  
\label{figure:lumped}
\end{figure}

% E: %FIXME: re-run with new rate estimates; add panel letters?
\begin{figure}
   % \centering 
    %\includegraphics[width=0.5\textwidth]{fig4a} \\ [40pt] 
       % \includegraphics[width=0.9\textwidth]{fig4b}
    \caption{
        Contributions of the two pathways to polyploidy.
        The one-step pathway is direct ID$\rightarrow$CP transitions.
        The two-step pathway consists of ID$\rightarrow$CD$\rightarrow$CP transitions.
        When considering only rates of transitions among the states (ignoring the diversification rate parameters), the one-step pathway dominates on short timescales and the two-step on long timescales (left panels).
        When also considering diversification within each state, the one-step pathway, in which polyploidization breaks down SI, dominates over any timescale (right panels).
        The top panels show the separate contributions of each pathway.
        The bottom panels show the proportional contribution of the one-step pathway (\ie one-step / [one-step + two-step]).
    }
    \label{figure:pathways}
\end{figure}

